<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Additional Usage of Statomatic</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Additional Usage of Statomatic</h1>



<p>This vignette will demonstrate additional usage of the statomatic
package beyond that shown in the basic workflow vignette. It is
recommended to read the basic workflow vignette before reading this
one.</p>
<div id="begin-by-loading-the-statomatic-package" class="section level3">
<h3>Begin by loading the statomatic package</h3>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(statomatic)</span></code></pre></div>
</div>
<div id="a-general-overview" class="section level3">
<h3>A general overview</h3>
<p>In this example we’ll demonstate how to use statomatic without
creating a statomatic data set.</p>
<p>All the functions used by statomatic were first developed to be
compatible with data.frames. The statomatic data set was added later as
an optional way to keep data and results organized all in one place. The
data.frame compatibility remains, and we can use most of statomatic’s
functions individually.</p>
<p>We’ll use the same normalized counts that were used in the
multi-group multi-factor example of the basic workflow vignette.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;normalized_counts.csv&quot;</span>)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">head</span>(x)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="co">#&gt;                    X Sample_600 Sample_601 Sample_602 Sample_616 Sample_617</span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="co">#&gt; 1 ENSMUSG00000051285  169.21935  309.83231 417.709088  214.03944  202.50044</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="co">#&gt; 2 ENSMUSG00000057363   47.53353   66.64997  54.805802   41.23696   49.30862</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="co">#&gt; 3 ENSMUSG00000100555   23.76676   55.84187   0.000000   48.10978   18.19153</span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="co">#&gt; 4 ENSMUSG00000078184   50.38554   59.44457   5.924952   12.76382   24.41495</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="co">#&gt; 5 ENSMUSG00000061024  273.79311  293.62015 232.554350  144.32935  360.00079</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a><span class="co">#&gt; 6 ENSMUSG00000025911  154.95929  525.99438 309.578721  208.14845  256.59631</span></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a><span class="co">#&gt;   Sample_603 Sample_604 Sample_605 Sample_606 Sample_621 Sample_607 Sample_608</span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a><span class="co">#&gt; 1  110.24045  171.38160   61.79141 190.642528  427.33305  228.73359  410.89116</span></span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a><span class="co">#&gt; 2   57.16171    0.00000    0.00000 103.558904   58.42021   63.76202   88.17407</span></span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a><span class="co">#&gt; 3   28.58086  151.21906   68.65713   5.884029   30.29196   21.25401   15.87133</span></span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a><span class="co">#&gt; 4   32.66384   10.08127    0.00000  44.718618   37.86495   19.22981   18.51655</span></span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a><span class="co">#&gt; 5  145.62627   55.44699  109.85140 222.416283  315.90190  253.02388  319.19013</span></span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a><span class="co">#&gt; 6  155.15322  171.38160    0.00000 264.781289  529.02750  301.60446  336.82494</span></span>
<span id="cb2-17"><a href="#cb2-17" tabindex="-1"></a><span class="co">#&gt;   Sample_609 Sample_610 Sample_623 Sample_611 Sample_612 Sample_613 Sample_614</span></span>
<span id="cb2-18"><a href="#cb2-18" tabindex="-1"></a><span class="co">#&gt; 1  313.05044  412.25499  133.76758  93.303195 248.040131   46.21469  417.35080</span></span>
<span id="cb2-19"><a href="#cb2-19" tabindex="-1"></a><span class="co">#&gt; 2   60.75650   14.45406   99.46820 103.366522  18.445254    0.00000   65.25144</span></span>
<span id="cb2-20"><a href="#cb2-20" tabindex="-1"></a><span class="co">#&gt; 3   45.30993   40.01052   30.86944   5.093783   1.456204  390.25738    0.00000</span></span>
<span id="cb2-21"><a href="#cb2-21" tabindex="-1"></a><span class="co">#&gt; 4   39.13131   36.86833   56.59398  57.025521  49.996347    0.00000   82.04884</span></span>
<span id="cb2-22"><a href="#cb2-22" tabindex="-1"></a><span class="co">#&gt; 5  209.04355  211.36447  221.23100 240.525945 248.040131  302.96296  440.60874</span></span>
<span id="cb2-23"><a href="#cb2-23" tabindex="-1"></a><span class="co">#&gt; 6  375.86649  528.09696  265.82019 148.589375 171.832106   10.26993  323.02694</span></span>
<span id="cb2-24"><a href="#cb2-24" tabindex="-1"></a><span class="co">#&gt;   Sample_615</span></span>
<span id="cb2-25"><a href="#cb2-25" tabindex="-1"></a><span class="co">#&gt; 1  513.48498</span></span>
<span id="cb2-26"><a href="#cb2-26" tabindex="-1"></a><span class="co">#&gt; 2   39.36603</span></span>
<span id="cb2-27"><a href="#cb2-27" tabindex="-1"></a><span class="co">#&gt; 3    0.00000</span></span>
<span id="cb2-28"><a href="#cb2-28" tabindex="-1"></a><span class="co">#&gt; 4   14.61838</span></span>
<span id="cb2-29"><a href="#cb2-29" tabindex="-1"></a><span class="co">#&gt; 5  213.52043</span></span>
<span id="cb2-30"><a href="#cb2-30" tabindex="-1"></a><span class="co">#&gt; 6  646.77697</span></span></code></pre></div>
<p>We always need to make sure that the data inside the data.frame is
purely numeric, and purely the numbers we’re trying to analyze. We can
use the cf() function to set a specified column (defaults to column 1)
as the rownames of the data.frame.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">cf</span>(x)</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="fu">head</span>(x)</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="co">#&gt;                    Sample_600 Sample_601 Sample_602 Sample_616 Sample_617</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051285  169.21935  309.83231 417.709088  214.03944  202.50044</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057363   47.53353   66.64997  54.805802   41.23696   49.30862</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000100555   23.76676   55.84187   0.000000   48.10978   18.19153</span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000078184   50.38554   59.44457   5.924952   12.76382   24.41495</span></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000061024  273.79311  293.62015 232.554350  144.32935  360.00079</span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025911  154.95929  525.99438 309.578721  208.14845  256.59631</span></span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a><span class="co">#&gt;                    Sample_603 Sample_604 Sample_605 Sample_606 Sample_621</span></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051285  110.24045  171.38160   61.79141 190.642528  427.33305</span></span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057363   57.16171    0.00000    0.00000 103.558904   58.42021</span></span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000100555   28.58086  151.21906   68.65713   5.884029   30.29196</span></span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000078184   32.66384   10.08127    0.00000  44.718618   37.86495</span></span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000061024  145.62627   55.44699  109.85140 222.416283  315.90190</span></span>
<span id="cb3-16"><a href="#cb3-16" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025911  155.15322  171.38160    0.00000 264.781289  529.02750</span></span>
<span id="cb3-17"><a href="#cb3-17" tabindex="-1"></a><span class="co">#&gt;                    Sample_607 Sample_608 Sample_609 Sample_610 Sample_623</span></span>
<span id="cb3-18"><a href="#cb3-18" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051285  228.73359  410.89116  313.05044  412.25499  133.76758</span></span>
<span id="cb3-19"><a href="#cb3-19" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057363   63.76202   88.17407   60.75650   14.45406   99.46820</span></span>
<span id="cb3-20"><a href="#cb3-20" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000100555   21.25401   15.87133   45.30993   40.01052   30.86944</span></span>
<span id="cb3-21"><a href="#cb3-21" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000078184   19.22981   18.51655   39.13131   36.86833   56.59398</span></span>
<span id="cb3-22"><a href="#cb3-22" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000061024  253.02388  319.19013  209.04355  211.36447  221.23100</span></span>
<span id="cb3-23"><a href="#cb3-23" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025911  301.60446  336.82494  375.86649  528.09696  265.82019</span></span>
<span id="cb3-24"><a href="#cb3-24" tabindex="-1"></a><span class="co">#&gt;                    Sample_611 Sample_612 Sample_613 Sample_614 Sample_615</span></span>
<span id="cb3-25"><a href="#cb3-25" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051285  93.303195 248.040131   46.21469  417.35080  513.48498</span></span>
<span id="cb3-26"><a href="#cb3-26" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057363 103.366522  18.445254    0.00000   65.25144   39.36603</span></span>
<span id="cb3-27"><a href="#cb3-27" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000100555   5.093783   1.456204  390.25738    0.00000    0.00000</span></span>
<span id="cb3-28"><a href="#cb3-28" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000078184  57.025521  49.996347    0.00000   82.04884   14.61838</span></span>
<span id="cb3-29"><a href="#cb3-29" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000061024 240.525945 248.040131  302.96296  440.60874  213.52043</span></span>
<span id="cb3-30"><a href="#cb3-30" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025911 148.589375 171.832106   10.26993  323.02694  646.77697</span></span></code></pre></div>
<p>We also always need a sample_info object to map our samples (columns
of the data.frame) to the experimental factors (groups, etc.).</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>sample_info <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">samples =</span> <span class="fu">colnames</span>(x), <span class="at">sex =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;Male&quot;</span>, <span class="dv">10</span>), <span class="fu">rep</span>(<span class="st">&quot;Female&quot;</span>, <span class="dv">10</span>)), </span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>                          <span class="at">genotype =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;WT&quot;</span>, <span class="dv">5</span>), <span class="fu">rep</span>(<span class="st">&quot;KO&quot;</span>, <span class="dv">5</span>), <span class="fu">rep</span>(<span class="st">&quot;WT&quot;</span>, <span class="dv">5</span>), <span class="fu">rep</span>(<span class="st">&quot;KO&quot;</span>, <span class="dv">5</span>)), </span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>                          <span class="at">group =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;M_WT&quot;</span>, <span class="dv">5</span>), <span class="fu">rep</span>(<span class="st">&quot;M_KO&quot;</span>, <span class="dv">5</span>), <span class="fu">rep</span>(<span class="st">&quot;F_WT&quot;</span>, <span class="dv">5</span>), <span class="fu">rep</span>(<span class="st">&quot;F_KO&quot;</span>, <span class="dv">5</span>)))</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>sample_info</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a><span class="co">#&gt;       samples    sex genotype group</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a><span class="co">#&gt; 1  Sample_600   Male       WT  M_WT</span></span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a><span class="co">#&gt; 2  Sample_601   Male       WT  M_WT</span></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a><span class="co">#&gt; 3  Sample_602   Male       WT  M_WT</span></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a><span class="co">#&gt; 4  Sample_616   Male       WT  M_WT</span></span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a><span class="co">#&gt; 5  Sample_617   Male       WT  M_WT</span></span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a><span class="co">#&gt; 6  Sample_603   Male       KO  M_KO</span></span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a><span class="co">#&gt; 7  Sample_604   Male       KO  M_KO</span></span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a><span class="co">#&gt; 8  Sample_605   Male       KO  M_KO</span></span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a><span class="co">#&gt; 9  Sample_606   Male       KO  M_KO</span></span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a><span class="co">#&gt; 10 Sample_621   Male       KO  M_KO</span></span>
<span id="cb4-16"><a href="#cb4-16" tabindex="-1"></a><span class="co">#&gt; 11 Sample_607 Female       WT  F_WT</span></span>
<span id="cb4-17"><a href="#cb4-17" tabindex="-1"></a><span class="co">#&gt; 12 Sample_608 Female       WT  F_WT</span></span>
<span id="cb4-18"><a href="#cb4-18" tabindex="-1"></a><span class="co">#&gt; 13 Sample_609 Female       WT  F_WT</span></span>
<span id="cb4-19"><a href="#cb4-19" tabindex="-1"></a><span class="co">#&gt; 14 Sample_610 Female       WT  F_WT</span></span>
<span id="cb4-20"><a href="#cb4-20" tabindex="-1"></a><span class="co">#&gt; 15 Sample_623 Female       WT  F_WT</span></span>
<span id="cb4-21"><a href="#cb4-21" tabindex="-1"></a><span class="co">#&gt; 16 Sample_611 Female       KO  F_KO</span></span>
<span id="cb4-22"><a href="#cb4-22" tabindex="-1"></a><span class="co">#&gt; 17 Sample_612 Female       KO  F_KO</span></span>
<span id="cb4-23"><a href="#cb4-23" tabindex="-1"></a><span class="co">#&gt; 18 Sample_613 Female       KO  F_KO</span></span>
<span id="cb4-24"><a href="#cb4-24" tabindex="-1"></a><span class="co">#&gt; 19 Sample_614 Female       KO  F_KO</span></span>
<span id="cb4-25"><a href="#cb4-25" tabindex="-1"></a><span class="co">#&gt; 20 Sample_615 Female       KO  F_KO</span></span></code></pre></div>
<p>Now, instead of building the statomatic data set, we can just jump
right in to the analysis. To do all the same things that sds_analyze
does, we can use the multigroup_main() function. This function will sort
each row of the data based on it’s distributions and scedasticity and
perform the applicable tests. The return value of the function is a
list, which contains all the results of the analysis.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">multigroup_main</span>(x, <span class="at">var1 =</span> sample_info<span class="sc">$</span>sex, <span class="at">var2 =</span> sample_info<span class="sc">$</span>genotype, <span class="at">var3 =</span> sample_info<span class="sc">$</span>group)</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="co">#&gt; sorting data 100% done...</span></span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="co">#&gt; Using two way anova </span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="co">#&gt; anova 100% done</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a><span class="co">#&gt; Tukey test 100% done</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a><span class="co">#&gt; Welch test 100% done</span></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a><span class="co">#&gt; DunnettT3 test 100% done</span></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a><span class="co">#&gt; Kruskal-Wallis Test 100% done</span></span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a><span class="co">#&gt; Dunn Test 100% done</span></span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a><span class="fu">names</span>(res)</span>
<span id="cb5-14"><a href="#cb5-14" tabindex="-1"></a><span class="co">#&gt; [1] &quot;anova_results&quot; &quot;welch_results&quot; &quot;kw_results&quot;    &quot;foldchanges&quot;  </span></span>
<span id="cb5-15"><a href="#cb5-15" tabindex="-1"></a><span class="co">#&gt; [5] &quot;ne&quot;            &quot;nu&quot;            &quot;nn&quot;</span></span></code></pre></div>
<p>You can access the elements of this list either by doing something
like res[[1]] (double brackets recommended) or res$anova_results. Also
notice that instead of specifying a design like we did when building a
statomatic data set, we used the var1, 2, and 3 to tell statomatic what
the experimental factors are. Based on these, statomatic knew to use a
two way anova.</p>
<p>The multigroup_main function can handle both a multi factor and a
single factor analysis. We could also do this:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">multigroup_main</span>(x, <span class="at">var1 =</span> sample_info<span class="sc">$</span>group)</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="co">#&gt; sorting data 100% done...</span></span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="co">#&gt; Using one way anova </span></span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a><span class="co">#&gt; anova 100% done</span></span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a><span class="co">#&gt; Tukey test 100% done</span></span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a><span class="co">#&gt; Welch test 100% done</span></span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a><span class="co">#&gt; DunnettT3 test 100% done</span></span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a><span class="co">#&gt; Kruskal-Wallis Test 100% done</span></span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a><span class="co">#&gt; Dunn Test 100% done</span></span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb6-13"><a href="#cb6-13" tabindex="-1"></a><span class="fu">names</span>(res)</span>
<span id="cb6-14"><a href="#cb6-14" tabindex="-1"></a><span class="co">#&gt; [1] &quot;anova_results&quot; &quot;welch_results&quot; &quot;kw_results&quot;    &quot;foldchanges&quot;  </span></span>
<span id="cb6-15"><a href="#cb6-15" tabindex="-1"></a><span class="co">#&gt; [5] &quot;ne&quot;            &quot;nu&quot;            &quot;nn&quot;</span></span></code></pre></div>
<p>And notice that now statomatic used a one way anova.</p>
<p>There is also a twogroup_main function, which works very similarly to
the example directly above. Note that the twogroup_main function can
only handle one factor with exactly two groups.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">twogroup_main</span>(x, <span class="at">var =</span> sample_info<span class="sc">$</span>sex)</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="co">#&gt; sorting data 100% done...</span></span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="co">#&gt; t-test 100% done</span></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a><span class="co">#&gt; Welch test 100% done</span></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a><span class="co">#&gt; Wilcox-test 100% done</span></span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a><span class="fu">names</span>(res)</span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a><span class="co">#&gt; [1] &quot;ttest_results&quot;  &quot;welch_results&quot;  &quot;wilcox_results&quot; &quot;foldchanges&quot;   </span></span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a><span class="co">#&gt; [5] &quot;ne&quot;             &quot;nu&quot;             &quot;nn&quot;</span></span></code></pre></div>
<p>The sds_analyze function is comprised of these two main functions,
along with some control flow stuff to make sure the data is routed to
the correct one. In the same way that sds_analyze is comprised of the
multigroup_main and twogroup_main functions above, each of the two main
functions is comprised of yet more functions which can also be used
independently.</p>
<p>We’ll first dig into the multigroup_main function.</p>
<p>By doing this:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>multigroup_main</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="co">#&gt; function (x, var1, var2 = NULL, var3 = NULL, padj = FALSE, write_files = FALSE, </span></span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a><span class="co">#&gt;     file_names = c(&quot;anova_results.csv&quot;, &quot;welch_results.csv&quot;, </span></span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a><span class="co">#&gt;         &quot;kw_results.csv&quot;)) </span></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a><span class="co">#&gt; {</span></span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a><span class="co">#&gt;     if (write_files == TRUE &amp;&amp; length(file_names) != 3) </span></span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a><span class="co">#&gt;         stop(&quot;This function creates three files but has not been provided three names&quot;)</span></span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a><span class="co">#&gt;     if (is.null(var3)) {</span></span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a><span class="co">#&gt;         var3 &lt;- var1</span></span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a><span class="co">#&gt;     }</span></span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a><span class="co">#&gt;     nenunn &lt;- test_norm(x, var1, var2)</span></span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a><span class="co">#&gt;     ne &lt;- nenunn[[1]]</span></span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a><span class="co">#&gt;     nu &lt;- nenunn[[2]]</span></span>
<span id="cb8-14"><a href="#cb8-14" tabindex="-1"></a><span class="co">#&gt;     nn &lt;- nenunn[[3]]</span></span>
<span id="cb8-15"><a href="#cb8-15" tabindex="-1"></a><span class="co">#&gt;     all_fcs &lt;- NULL</span></span>
<span id="cb8-16"><a href="#cb8-16" tabindex="-1"></a><span class="co">#&gt;     anova_results &lt;- data.frame()</span></span>
<span id="cb8-17"><a href="#cb8-17" tabindex="-1"></a><span class="co">#&gt;     welch_results &lt;- data.frame()</span></span>
<span id="cb8-18"><a href="#cb8-18" tabindex="-1"></a><span class="co">#&gt;     kw_results &lt;- data.frame()</span></span>
<span id="cb8-19"><a href="#cb8-19" tabindex="-1"></a><span class="co">#&gt;     if (nrow(ne) &gt; 0) {</span></span>
<span id="cb8-20"><a href="#cb8-20" tabindex="-1"></a><span class="co">#&gt;         my_anova &lt;- run_anova(ne, var1 = var1, var2 = var2, padj)</span></span>
<span id="cb8-21"><a href="#cb8-21" tabindex="-1"></a><span class="co">#&gt;         if (padj == TRUE) {</span></span>
<span id="cb8-22"><a href="#cb8-22" tabindex="-1"></a><span class="co">#&gt;             if (is.null(var2)) {</span></span>
<span id="cb8-23"><a href="#cb8-23" tabindex="-1"></a><span class="co">#&gt;                 colnames(my_anova) &lt;- c(substitute(var1), paste0(deparse(substitute(var1)), </span></span>
<span id="cb8-24"><a href="#cb8-24" tabindex="-1"></a><span class="co">#&gt;                   &quot;_padj&quot;))</span></span>
<span id="cb8-25"><a href="#cb8-25" tabindex="-1"></a><span class="co">#&gt;             }</span></span>
<span id="cb8-26"><a href="#cb8-26" tabindex="-1"></a><span class="co">#&gt;             else {</span></span>
<span id="cb8-27"><a href="#cb8-27" tabindex="-1"></a><span class="co">#&gt;                 colnames(my_anova) &lt;- c(substitute(var1), substitute(var2), </span></span>
<span id="cb8-28"><a href="#cb8-28" tabindex="-1"></a><span class="co">#&gt;                   &quot;interaction&quot;, paste0(deparse(substitute(var1)), </span></span>
<span id="cb8-29"><a href="#cb8-29" tabindex="-1"></a><span class="co">#&gt;                     &quot;_padj&quot;), paste0(deparse(substitute(var2)), </span></span>
<span id="cb8-30"><a href="#cb8-30" tabindex="-1"></a><span class="co">#&gt;                     &quot;_padj&quot;), &quot;interaction_padj&quot;)</span></span>
<span id="cb8-31"><a href="#cb8-31" tabindex="-1"></a><span class="co">#&gt;             }</span></span>
<span id="cb8-32"><a href="#cb8-32" tabindex="-1"></a><span class="co">#&gt;         }</span></span>
<span id="cb8-33"><a href="#cb8-33" tabindex="-1"></a><span class="co">#&gt;         else {</span></span>
<span id="cb8-34"><a href="#cb8-34" tabindex="-1"></a><span class="co">#&gt;             if (is.null(var2)) {</span></span>
<span id="cb8-35"><a href="#cb8-35" tabindex="-1"></a><span class="co">#&gt;                 colnames(my_anova) &lt;- &quot;Pval&quot;</span></span>
<span id="cb8-36"><a href="#cb8-36" tabindex="-1"></a><span class="co">#&gt;             }</span></span>
<span id="cb8-37"><a href="#cb8-37" tabindex="-1"></a><span class="co">#&gt;             else {</span></span>
<span id="cb8-38"><a href="#cb8-38" tabindex="-1"></a><span class="co">#&gt;                 colnames(my_anova) &lt;- c(substitute(var1), substitute(var2), </span></span>
<span id="cb8-39"><a href="#cb8-39" tabindex="-1"></a><span class="co">#&gt;                   &quot;interaction&quot;)</span></span>
<span id="cb8-40"><a href="#cb8-40" tabindex="-1"></a><span class="co">#&gt;             }</span></span>
<span id="cb8-41"><a href="#cb8-41" tabindex="-1"></a><span class="co">#&gt;         }</span></span>
<span id="cb8-42"><a href="#cb8-42" tabindex="-1"></a><span class="co">#&gt;         tukey_res &lt;- run_tukey(ne, var3)</span></span>
<span id="cb8-43"><a href="#cb8-43" tabindex="-1"></a><span class="co">#&gt;         fc_ne &lt;- fold_change(ne, var3)</span></span>
<span id="cb8-44"><a href="#cb8-44" tabindex="-1"></a><span class="co">#&gt;         tukey_res &lt;- combine_fc_pvals(tukey_res, fc_ne)</span></span>
<span id="cb8-45"><a href="#cb8-45" tabindex="-1"></a><span class="co">#&gt;         anova_results &lt;- cbind(my_anova, tukey_res)</span></span>
<span id="cb8-46"><a href="#cb8-46" tabindex="-1"></a><span class="co">#&gt;         all_fcs &lt;- rbind(all_fcs, fc_ne)</span></span>
<span id="cb8-47"><a href="#cb8-47" tabindex="-1"></a><span class="co">#&gt;     }</span></span>
<span id="cb8-48"><a href="#cb8-48" tabindex="-1"></a><span class="co">#&gt;     if (nrow(nu) &gt; 0) {</span></span>
<span id="cb8-49"><a href="#cb8-49" tabindex="-1"></a><span class="co">#&gt;         welch_res &lt;- run_welch(nu, var3, padj)</span></span>
<span id="cb8-50"><a href="#cb8-50" tabindex="-1"></a><span class="co">#&gt;         dunnett_res &lt;- run_dunnett(nu, var3)</span></span>
<span id="cb8-51"><a href="#cb8-51" tabindex="-1"></a><span class="co">#&gt;         fc_nu &lt;- fold_change(nu, var3)</span></span>
<span id="cb8-52"><a href="#cb8-52" tabindex="-1"></a><span class="co">#&gt;         dunnett_res &lt;- combine_fc_pvals(dunnett_res, fc_nu)</span></span>
<span id="cb8-53"><a href="#cb8-53" tabindex="-1"></a><span class="co">#&gt;         welch_results &lt;- cbind(welch_res, dunnett_res)</span></span>
<span id="cb8-54"><a href="#cb8-54" tabindex="-1"></a><span class="co">#&gt;         all_fcs &lt;- rbind(all_fcs, fc_nu)</span></span>
<span id="cb8-55"><a href="#cb8-55" tabindex="-1"></a><span class="co">#&gt;     }</span></span>
<span id="cb8-56"><a href="#cb8-56" tabindex="-1"></a><span class="co">#&gt;     if (nrow(nn) &gt; 0) {</span></span>
<span id="cb8-57"><a href="#cb8-57" tabindex="-1"></a><span class="co">#&gt;         krusk_res &lt;- run_kruskal(nn, var3, padj)</span></span>
<span id="cb8-58"><a href="#cb8-58" tabindex="-1"></a><span class="co">#&gt;         dunn_res &lt;- run_dunn(nn, var3)</span></span>
<span id="cb8-59"><a href="#cb8-59" tabindex="-1"></a><span class="co">#&gt;         fc_nn &lt;- fold_change(nn, var3)</span></span>
<span id="cb8-60"><a href="#cb8-60" tabindex="-1"></a><span class="co">#&gt;         dunn_res &lt;- combine_fc_pvals(dunn_res, fc_nn)</span></span>
<span id="cb8-61"><a href="#cb8-61" tabindex="-1"></a><span class="co">#&gt;         kw_results &lt;- cbind(krusk_res, dunn_res)</span></span>
<span id="cb8-62"><a href="#cb8-62" tabindex="-1"></a><span class="co">#&gt;         all_fcs &lt;- rbind(all_fcs, fc_nn)</span></span>
<span id="cb8-63"><a href="#cb8-63" tabindex="-1"></a><span class="co">#&gt;     }</span></span>
<span id="cb8-64"><a href="#cb8-64" tabindex="-1"></a><span class="co">#&gt;     if (write_files == TRUE) {</span></span>
<span id="cb8-65"><a href="#cb8-65" tabindex="-1"></a><span class="co">#&gt;         write.csv(anova_results, file_names[1])</span></span>
<span id="cb8-66"><a href="#cb8-66" tabindex="-1"></a><span class="co">#&gt;         write.csv(welch_results, file_names[2])</span></span>
<span id="cb8-67"><a href="#cb8-67" tabindex="-1"></a><span class="co">#&gt;         write.csv(kw_results, file_names[3])</span></span>
<span id="cb8-68"><a href="#cb8-68" tabindex="-1"></a><span class="co">#&gt;         return(NULL)</span></span>
<span id="cb8-69"><a href="#cb8-69" tabindex="-1"></a><span class="co">#&gt;     }</span></span>
<span id="cb8-70"><a href="#cb8-70" tabindex="-1"></a><span class="co">#&gt;     res &lt;- list(anova_results = anova_results, welch_results = welch_results, </span></span>
<span id="cb8-71"><a href="#cb8-71" tabindex="-1"></a><span class="co">#&gt;         kw_results = kw_results, foldchanges = all_fcs, ne = ne, </span></span>
<span id="cb8-72"><a href="#cb8-72" tabindex="-1"></a><span class="co">#&gt;         nu = nu, nn = nn)</span></span>
<span id="cb8-73"><a href="#cb8-73" tabindex="-1"></a><span class="co">#&gt;     return(res)</span></span>
<span id="cb8-74"><a href="#cb8-74" tabindex="-1"></a><span class="co">#&gt; }</span></span>
<span id="cb8-75"><a href="#cb8-75" tabindex="-1"></a><span class="co">#&gt; &lt;bytecode: 0x158f7fc70&gt;</span></span>
<span id="cb8-76"><a href="#cb8-76" tabindex="-1"></a><span class="co">#&gt; &lt;environment: namespace:statomatic&gt;</span></span></code></pre></div>
<p>We can see the source code of the function.</p>
<p>Don’t worry about the whole thing, but notice the functions:
test_norm, run_anova, run_tukey, run_welch, run_dunnett, run_kruskal,
run_dunn, and fold_change.</p>
<p>Each of these functions is available to use independently of the
rest.</p>
<p>The test_norm function is the function that sorts the data by
distribution and scedasticity. It can handle either 1 or 2 factors.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">test_norm</span>(x, sample_info<span class="sc">$</span>sex, sample_info<span class="sc">$</span>genotype)</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a><span class="co">#&gt; sorting data 100% done...</span></span></code></pre></div>
<p>The return value is a list with three data.frames. The first,
res[[1]], is data determined to have normally distributed residuals and
equal variances amongst the factors. The second, res[[2]], has normally
distributed residuals but unequal variances amongst the factors. The
third, res[[3]], is data with non-normally distributed residuals.</p>
<p>To use test_norm with only one factor, simply supply only one.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">test_norm</span>(x, sample_info<span class="sc">$</span>group)</span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a><span class="co">#&gt; sorting data 100% done...</span></span></code></pre></div>
<p>The usage of the rest of the functions is identical to that of the
test_norm function, and their names are descriptive of what they do.</p>
<p>run_anova runs an anova test, run_tukey runs a tukey test, etc.</p>
<p>The run_anova function can handle either one factor or two, and
either a one way or two way anova will be performed in accordance with
the number of factors supplied. The rest of the functions only support
using one factor.</p>
<p>We’ll demonstate the run_anova function.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>anova_res <span class="ot">&lt;-</span> <span class="fu">run_anova</span>(x, sample_info<span class="sc">$</span>sex, sample_info<span class="sc">$</span>genotype)</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a><span class="co">#&gt; Using two way anova </span></span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a><span class="co">#&gt; anova 100% done</span></span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a><span class="fu">head</span>(anova_res)</span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a><span class="co">#&gt;                    sample_info$sex sample_info$genotype interaction</span></span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051285       0.4181785            0.4267951  0.79596064</span></span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057363       0.6363190            0.3757298  0.70346317</span></span>
<span id="cb11-8"><a href="#cb11-8" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000100555       0.7760540            0.3688330  0.80312281</span></span>
<span id="cb11-9"><a href="#cb11-9" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000078184       0.3813036            0.9576038  0.57458343</span></span>
<span id="cb11-10"><a href="#cb11-10" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000061024       0.1879463            0.5524404  0.08030823</span></span>
<span id="cb11-11"><a href="#cb11-11" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025911       0.5146527            0.3078191  0.83172345</span></span></code></pre></div>
<p>And the run_tukey function.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a>tukey_res <span class="ot">&lt;-</span> <span class="fu">run_tukey</span>(x, sample_info<span class="sc">$</span>group)</span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a><span class="co">#&gt; Tukey test 100% done...</span></span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a><span class="fu">head</span>(tukey_res)</span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a><span class="co">#&gt;                    Pval_F_KO_vs_F_WT Pval_F_KO_vs_M_KO Pval_F_KO_vs_M_WT</span></span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051285         0.9790520         0.8652082         0.9999995</span></span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057363         0.7955281         0.9998882         0.9899203</span></span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000100555         0.8379946         0.9800907         0.8257741</span></span>
<span id="cb12-8"><a href="#cb12-8" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000078184         0.9699575         0.7280883         0.9050970</span></span>
<span id="cb12-9"><a href="#cb12-9" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000061024         0.8094858         0.1413002         0.9469754</span></span>
<span id="cb12-10"><a href="#cb12-10" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025911         0.8062227         0.9883949         0.9925533</span></span>
<span id="cb12-11"><a href="#cb12-11" tabindex="-1"></a><span class="co">#&gt;                    Pval_F_WT_vs_M_KO Pval_F_WT_vs_M_WT Pval_M_KO_vs_M_WT</span></span>
<span id="cb12-12"><a href="#cb12-12" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051285         0.6568296         0.9773126         0.8699999</span></span>
<span id="cb12-13"><a href="#cb12-13" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057363         0.7597698         0.9258447         0.9820468</span></span>
<span id="cb12-14"><a href="#cb12-14" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000100555         0.9688062         0.9999939         0.9635951</span></span>
<span id="cb12-15"><a href="#cb12-15" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000078184         0.9310571         0.9954529         0.9824965</span></span>
<span id="cb12-16"><a href="#cb12-16" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000061024         0.5161829         0.9850262         0.3321566</span></span>
<span id="cb12-17"><a href="#cb12-17" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025911         0.6260176         0.9228793         0.9330056</span></span>
<span id="cb12-18"><a href="#cb12-18" tabindex="-1"></a><span class="co">#&gt;                          test</span></span>
<span id="cb12-19"><a href="#cb12-19" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051285 tukey-test</span></span>
<span id="cb12-20"><a href="#cb12-20" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057363 tukey-test</span></span>
<span id="cb12-21"><a href="#cb12-21" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000100555 tukey-test</span></span>
<span id="cb12-22"><a href="#cb12-22" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000078184 tukey-test</span></span>
<span id="cb12-23"><a href="#cb12-23" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000061024 tukey-test</span></span>
<span id="cb12-24"><a href="#cb12-24" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025911 tukey-test</span></span></code></pre></div>
<p>We’ll discuss only briefly about the twogroup_main function. You can
view it’s source code the same way we did for the multigroup_main
function.</p>
<p>This function is made up of the same test_norm function, run_ttest,
run_welch, and run_wilcox, which are all available to use
independently.</p>
<p>Run_ttest runs a t-test, run welch runs a welch test (and is the same
welch test used in the multigroup case), and run_wilcox runs a wilcox
test. Each of these three functions can handle only one factor, and
their usage is identical to the run_tukey function.</p>
<p>To end this vignette, we’ll demonstrate a more advanced use case for
using the multigroup_main function.</p>
</div>
<div id="a-more-advanced-example" class="section level3">
<h3>A more advanced example</h3>
<p>In this example we have six different tables we want to analyze.
These tables are from a metagenomic analysis, and contain abundance
information at the phylum, class, order, family, genus, and species
level. We looked at this species data before, in the basic workflow
vignette.</p>
<p>There are 24 samples in this experiment, and 3 experiemental factors,
with 8 samples per factor. We will call the factors groups 1-3.
Therefore, we’ll use the multigroup main function with one factor.</p>
<p>We’re going to apply the statomatic method to all six of these tables
in order to identify differences between groups at each taxonomic
level.</p>
<p>We read in the data all together and store it in a list.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="co">#get all files in the current directory with abundance.csv in the name </span></span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">&quot;.&quot;</span>, <span class="at">pattern =</span> <span class="st">&quot;abundance.csv&quot;</span>)</span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a>a</span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a><span class="co">#&gt; [1] &quot;class_abundance.csv&quot;   &quot;family_abundance.csv&quot;  &quot;genus_abundance.csv&quot;  </span></span>
<span id="cb13-5"><a href="#cb13-5" tabindex="-1"></a><span class="co">#&gt; [4] &quot;order_abundance.csv&quot;   &quot;phylum_abundance.csv&quot;  &quot;species_abundance.csv&quot;</span></span></code></pre></div>
<p>Notice that the file names are in alphabetical order.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="co">#read them in with read.csv</span></span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a>read_in <span class="ot">&lt;-</span> <span class="fu">lapply</span>(a, read.csv)</span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a><span class="co">#look at first three rows of data</span></span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a><span class="fu">lapply</span>(read_in, head, <span class="dv">3</span>)</span>
<span id="cb14-5"><a href="#cb14-5" tabindex="-1"></a><span class="co">#&gt; [[1]]</span></span>
<span id="cb14-6"><a href="#cb14-6" tabindex="-1"></a><span class="co">#&gt;              class sample_1 sample_2 sample_3 sample_4 sample_5 sample_6</span></span>
<span id="cb14-7"><a href="#cb14-7" tabindex="-1"></a><span class="co">#&gt; 1       Clostridia      102      475     2910     1188     1278      324</span></span>
<span id="cb14-8"><a href="#cb14-8" tabindex="-1"></a><span class="co">#&gt; 2      Bacteroidia     1066     1005      383      801      623      628</span></span>
<span id="cb14-9"><a href="#cb14-9" tabindex="-1"></a><span class="co">#&gt; 3 Erysipelotrichia      588      186      141      157      127      123</span></span>
<span id="cb14-10"><a href="#cb14-10" tabindex="-1"></a><span class="co">#&gt;   sample_7 sample_8 sample_9 sample_10 sample_11 sample_12 sample_13 sample_14</span></span>
<span id="cb14-11"><a href="#cb14-11" tabindex="-1"></a><span class="co">#&gt; 1     4350      641     1176       883       426      5880        82       158</span></span>
<span id="cb14-12"><a href="#cb14-12" tabindex="-1"></a><span class="co">#&gt; 2      596      549      259       264       551       270       207       582</span></span>
<span id="cb14-13"><a href="#cb14-13" tabindex="-1"></a><span class="co">#&gt; 3      752      139      129       132       229       249       131       464</span></span>
<span id="cb14-14"><a href="#cb14-14" tabindex="-1"></a><span class="co">#&gt;   sample_15 sample_16 sample_17 sample_18 sample_19 sample_20 sample_21</span></span>
<span id="cb14-15"><a href="#cb14-15" tabindex="-1"></a><span class="co">#&gt; 1      1632       531       384       829      1708      4864       799</span></span>
<span id="cb14-16"><a href="#cb14-16" tabindex="-1"></a><span class="co">#&gt; 2       972       700       477       724      1227       661      1754</span></span>
<span id="cb14-17"><a href="#cb14-17" tabindex="-1"></a><span class="co">#&gt; 3       144       135       121       746       752       171       251</span></span>
<span id="cb14-18"><a href="#cb14-18" tabindex="-1"></a><span class="co">#&gt;   sample_22 sample_23 sample_24</span></span>
<span id="cb14-19"><a href="#cb14-19" tabindex="-1"></a><span class="co">#&gt; 1       841      1262       524</span></span>
<span id="cb14-20"><a href="#cb14-20" tabindex="-1"></a><span class="co">#&gt; 2      1656       688      1523</span></span>
<span id="cb14-21"><a href="#cb14-21" tabindex="-1"></a><span class="co">#&gt; 3       230       410        62</span></span>
<span id="cb14-22"><a href="#cb14-22" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb14-23"><a href="#cb14-23" tabindex="-1"></a><span class="co">#&gt; [[2]]</span></span>
<span id="cb14-24"><a href="#cb14-24" tabindex="-1"></a><span class="co">#&gt;             family sample_1 sample_2 sample_3 sample_4 sample_5 sample_6</span></span>
<span id="cb14-25"><a href="#cb14-25" tabindex="-1"></a><span class="co">#&gt; 1  Lachnospiraceae       36      123     1888      426      586       69</span></span>
<span id="cb14-26"><a href="#cb14-26" tabindex="-1"></a><span class="co">#&gt; 2 Oscillospiraceae       39      215      501      487      344      145</span></span>
<span id="cb14-27"><a href="#cb14-27" tabindex="-1"></a><span class="co">#&gt; 3     unclassified       76      204      384      342      421       83</span></span>
<span id="cb14-28"><a href="#cb14-28" tabindex="-1"></a><span class="co">#&gt;   sample_7 sample_8 sample_9 sample_10 sample_11 sample_12 sample_13 sample_14</span></span>
<span id="cb14-29"><a href="#cb14-29" tabindex="-1"></a><span class="co">#&gt; 1     2786      258      808       266       217      3684        38        56</span></span>
<span id="cb14-30"><a href="#cb14-30" tabindex="-1"></a><span class="co">#&gt; 2     1066      255      175       324        66      1301        24        54</span></span>
<span id="cb14-31"><a href="#cb14-31" tabindex="-1"></a><span class="co">#&gt; 3      585      159      310       229        56      1308        45        67</span></span>
<span id="cb14-32"><a href="#cb14-32" tabindex="-1"></a><span class="co">#&gt;   sample_15 sample_16 sample_17 sample_18 sample_19 sample_20 sample_21</span></span>
<span id="cb14-33"><a href="#cb14-33" tabindex="-1"></a><span class="co">#&gt; 1      1099       395        39       136       764      3133       217</span></span>
<span id="cb14-34"><a href="#cb14-34" tabindex="-1"></a><span class="co">#&gt; 2       173        71        32       258       491       896       293</span></span>
<span id="cb14-35"><a href="#cb14-35" tabindex="-1"></a><span class="co">#&gt; 3       369       124        71       286       514      1188       273</span></span>
<span id="cb14-36"><a href="#cb14-36" tabindex="-1"></a><span class="co">#&gt;   sample_22 sample_23 sample_24</span></span>
<span id="cb14-37"><a href="#cb14-37" tabindex="-1"></a><span class="co">#&gt; 1       290       508        98</span></span>
<span id="cb14-38"><a href="#cb14-38" tabindex="-1"></a><span class="co">#&gt; 2       292       480       284</span></span>
<span id="cb14-39"><a href="#cb14-39" tabindex="-1"></a><span class="co">#&gt; 3       288       262       197</span></span>
<span id="cb14-40"><a href="#cb14-40" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb14-41"><a href="#cb14-41" tabindex="-1"></a><span class="co">#&gt; [[3]]</span></span>
<span id="cb14-42"><a href="#cb14-42" tabindex="-1"></a><span class="co">#&gt;          genus sample_1 sample_2 sample_3 sample_4 sample_5 sample_6 sample_7</span></span>
<span id="cb14-43"><a href="#cb14-43" tabindex="-1"></a><span class="co">#&gt; 1 unclassified      281      304     1902      564      688      235     2828</span></span>
<span id="cb14-44"><a href="#cb14-44" tabindex="-1"></a><span class="co">#&gt; 2   Dubosiella      583      181      118      145      112      118      665</span></span>
<span id="cb14-45"><a href="#cb14-45" tabindex="-1"></a><span class="co">#&gt; 3  Bacteroides      406       92       91      106      111      185      106</span></span>
<span id="cb14-46"><a href="#cb14-46" tabindex="-1"></a><span class="co">#&gt;   sample_8 sample_9 sample_10 sample_11 sample_12 sample_13 sample_14 sample_15</span></span>
<span id="cb14-47"><a href="#cb14-47" tabindex="-1"></a><span class="co">#&gt; 1      370      758       322       228      3085       114       152      1477</span></span>
<span id="cb14-48"><a href="#cb14-48" tabindex="-1"></a><span class="co">#&gt; 2      131      121       117       113       210       120       460       117</span></span>
<span id="cb14-49"><a href="#cb14-49" tabindex="-1"></a><span class="co">#&gt; 3      156       91        93        78        95        97       421       165</span></span>
<span id="cb14-50"><a href="#cb14-50" tabindex="-1"></a><span class="co">#&gt;   sample_16 sample_17 sample_18 sample_19 sample_20 sample_21 sample_22</span></span>
<span id="cb14-51"><a href="#cb14-51" tabindex="-1"></a><span class="co">#&gt; 1       611       185       510      1002      3544       611       665</span></span>
<span id="cb14-52"><a href="#cb14-52" tabindex="-1"></a><span class="co">#&gt; 2       105       114       737       728       123       245       220</span></span>
<span id="cb14-53"><a href="#cb14-53" tabindex="-1"></a><span class="co">#&gt; 3       129       102       129       128       170       617       677</span></span>
<span id="cb14-54"><a href="#cb14-54" tabindex="-1"></a><span class="co">#&gt;   sample_23 sample_24</span></span>
<span id="cb14-55"><a href="#cb14-55" tabindex="-1"></a><span class="co">#&gt; 1       717       323</span></span>
<span id="cb14-56"><a href="#cb14-56" tabindex="-1"></a><span class="co">#&gt; 2       402        57</span></span>
<span id="cb14-57"><a href="#cb14-57" tabindex="-1"></a><span class="co">#&gt; 3       140       774</span></span>
<span id="cb14-58"><a href="#cb14-58" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb14-59"><a href="#cb14-59" tabindex="-1"></a><span class="co">#&gt; [[4]]</span></span>
<span id="cb14-60"><a href="#cb14-60" tabindex="-1"></a><span class="co">#&gt;                order sample_1 sample_2 sample_3 sample_4 sample_5 sample_6</span></span>
<span id="cb14-61"><a href="#cb14-61" tabindex="-1"></a><span class="co">#&gt; 1      Eubacteriales      100      468     2888     1184     1261      318</span></span>
<span id="cb14-62"><a href="#cb14-62" tabindex="-1"></a><span class="co">#&gt; 2      Bacteroidales     1065     1005      380      800      621      626</span></span>
<span id="cb14-63"><a href="#cb14-63" tabindex="-1"></a><span class="co">#&gt; 3 Erysipelotrichales      588      186      141      157      127      123</span></span>
<span id="cb14-64"><a href="#cb14-64" tabindex="-1"></a><span class="co">#&gt;   sample_7 sample_8 sample_9 sample_10 sample_11 sample_12 sample_13 sample_14</span></span>
<span id="cb14-65"><a href="#cb14-65" tabindex="-1"></a><span class="co">#&gt; 1     4338      632     1167       874       426      5858        82       158</span></span>
<span id="cb14-66"><a href="#cb14-66" tabindex="-1"></a><span class="co">#&gt; 2      593      549      258       263       551       270       206       582</span></span>
<span id="cb14-67"><a href="#cb14-67" tabindex="-1"></a><span class="co">#&gt; 3      752      139      129       132       229       249       131       464</span></span>
<span id="cb14-68"><a href="#cb14-68" tabindex="-1"></a><span class="co">#&gt;   sample_15 sample_16 sample_17 sample_18 sample_19 sample_20 sample_21</span></span>
<span id="cb14-69"><a href="#cb14-69" tabindex="-1"></a><span class="co">#&gt; 1      1625       529       382       812      1696      4845       783</span></span>
<span id="cb14-70"><a href="#cb14-70" tabindex="-1"></a><span class="co">#&gt; 2       967       696       477       716      1224       656      1745</span></span>
<span id="cb14-71"><a href="#cb14-71" tabindex="-1"></a><span class="co">#&gt; 3       144       135       121       746       752       171       251</span></span>
<span id="cb14-72"><a href="#cb14-72" tabindex="-1"></a><span class="co">#&gt;   sample_22 sample_23 sample_24</span></span>
<span id="cb14-73"><a href="#cb14-73" tabindex="-1"></a><span class="co">#&gt; 1       824      1245       521</span></span>
<span id="cb14-74"><a href="#cb14-74" tabindex="-1"></a><span class="co">#&gt; 2      1650       687      1520</span></span>
<span id="cb14-75"><a href="#cb14-75" tabindex="-1"></a><span class="co">#&gt; 3       230       410        62</span></span>
<span id="cb14-76"><a href="#cb14-76" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb14-77"><a href="#cb14-77" tabindex="-1"></a><span class="co">#&gt; [[5]]</span></span>
<span id="cb14-78"><a href="#cb14-78" tabindex="-1"></a><span class="co">#&gt;           phylum sample_1 sample_2 sample_3 sample_4 sample_5 sample_6 sample_7</span></span>
<span id="cb14-79"><a href="#cb14-79" tabindex="-1"></a><span class="co">#&gt; 1      Bacillota      710      790     3214     1471     1573      495     5279</span></span>
<span id="cb14-80"><a href="#cb14-80" tabindex="-1"></a><span class="co">#&gt; 2   Bacteroidota     1071     1007      385      801      627      631      603</span></span>
<span id="cb14-81"><a href="#cb14-81" tabindex="-1"></a><span class="co">#&gt; 3 Pseudomonadota       14      269      475      372      396       29      197</span></span>
<span id="cb14-82"><a href="#cb14-82" tabindex="-1"></a><span class="co">#&gt;   sample_8 sample_9 sample_10 sample_11 sample_12 sample_13 sample_14 sample_15</span></span>
<span id="cb14-83"><a href="#cb14-83" tabindex="-1"></a><span class="co">#&gt; 1      836     1506      1117      1109      6689       234       668      1948</span></span>
<span id="cb14-84"><a href="#cb14-84" tabindex="-1"></a><span class="co">#&gt; 2      551      261       267       551       274       207       584       978</span></span>
<span id="cb14-85"><a href="#cb14-85" tabindex="-1"></a><span class="co">#&gt; 3      266      402       395       263       352       413        14       462</span></span>
<span id="cb14-86"><a href="#cb14-86" tabindex="-1"></a><span class="co">#&gt;   sample_16 sample_17 sample_18 sample_19 sample_20 sample_21 sample_22</span></span>
<span id="cb14-87"><a href="#cb14-87" tabindex="-1"></a><span class="co">#&gt; 1       712       999      1707      2770      5492      1231      1214</span></span>
<span id="cb14-88"><a href="#cb14-88" tabindex="-1"></a><span class="co">#&gt; 2       708       477       736      1235       670      1764      1671</span></span>
<span id="cb14-89"><a href="#cb14-89" tabindex="-1"></a><span class="co">#&gt; 3       176       409        25        35       516        35       163</span></span>
<span id="cb14-90"><a href="#cb14-90" tabindex="-1"></a><span class="co">#&gt;   sample_23 sample_24</span></span>
<span id="cb14-91"><a href="#cb14-91" tabindex="-1"></a><span class="co">#&gt; 1      1862       630</span></span>
<span id="cb14-92"><a href="#cb14-92" tabindex="-1"></a><span class="co">#&gt; 2       704      1532</span></span>
<span id="cb14-93"><a href="#cb14-93" tabindex="-1"></a><span class="co">#&gt; 3        29        18</span></span>
<span id="cb14-94"><a href="#cb14-94" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb14-95"><a href="#cb14-95" tabindex="-1"></a><span class="co">#&gt; [[6]]</span></span>
<span id="cb14-96"><a href="#cb14-96" tabindex="-1"></a><span class="co">#&gt;                            species sample_1 sample_2 sample_3 sample_4 sample_5</span></span>
<span id="cb14-97"><a href="#cb14-97" tabindex="-1"></a><span class="co">#&gt; 1        Lachnospiraceae bacterium       14       33      245      224      167</span></span>
<span id="cb14-98"><a href="#cb14-98" tabindex="-1"></a><span class="co">#&gt; 2          Dubosiella newyorkensis      583      181      118      145      112</span></span>
<span id="cb14-99"><a href="#cb14-99" tabindex="-1"></a><span class="co">#&gt; 3 Parasutterella excrementihominis        4      241      431      327      343</span></span>
<span id="cb14-100"><a href="#cb14-100" tabindex="-1"></a><span class="co">#&gt;   sample_6 sample_7 sample_8 sample_9 sample_10 sample_11 sample_12 sample_13</span></span>
<span id="cb14-101"><a href="#cb14-101" tabindex="-1"></a><span class="co">#&gt; 1        9      996       71      312        85        91      1731        22</span></span>
<span id="cb14-102"><a href="#cb14-102" tabindex="-1"></a><span class="co">#&gt; 2      118      665      131      121       117       113       210       120</span></span>
<span id="cb14-103"><a href="#cb14-103" tabindex="-1"></a><span class="co">#&gt; 3       21      128      241      367       364       246       255       393</span></span>
<span id="cb14-104"><a href="#cb14-104" tabindex="-1"></a><span class="co">#&gt;   sample_14 sample_15 sample_16 sample_17 sample_18 sample_19 sample_20</span></span>
<span id="cb14-105"><a href="#cb14-105" tabindex="-1"></a><span class="co">#&gt; 1        12       221       289        11        35       168      1907</span></span>
<span id="cb14-106"><a href="#cb14-106" tabindex="-1"></a><span class="co">#&gt; 2       460       117       105       114       737       728       123</span></span>
<span id="cb14-107"><a href="#cb14-107" tabindex="-1"></a><span class="co">#&gt; 3         8       386       120       385         7         4       433</span></span>
<span id="cb14-108"><a href="#cb14-108" tabindex="-1"></a><span class="co">#&gt;   sample_21 sample_22 sample_23 sample_24</span></span>
<span id="cb14-109"><a href="#cb14-109" tabindex="-1"></a><span class="co">#&gt; 1        42       103       136        28</span></span>
<span id="cb14-110"><a href="#cb14-110" tabindex="-1"></a><span class="co">#&gt; 2       245       220       402        57</span></span>
<span id="cb14-111"><a href="#cb14-111" tabindex="-1"></a><span class="co">#&gt; 3         4       117        10         1</span></span></code></pre></div>
<p>Each of these tables needs the cf function.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a><span class="co">#apply cf to each table</span></span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a>tables <span class="ot">&lt;-</span> <span class="fu">lapply</span>(read_in, cf)</span></code></pre></div>
<p>We can use the same sample info object for all of these tables, since
they all contain the same samples.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="co">#tables[[1]] is the first table in the tables list. all the tables have the same column names</span></span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a>sample_info <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">sample =</span> <span class="fu">colnames</span>(tables[[<span class="dv">1</span>]]), <span class="at">group =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;group1&quot;</span>, <span class="dv">8</span>), <span class="fu">rep</span>(<span class="st">&quot;group2&quot;</span>, <span class="dv">8</span>), <span class="fu">rep</span>(<span class="st">&quot;group3&quot;</span>, <span class="dv">8</span>)))</span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a>sample_info</span>
<span id="cb16-4"><a href="#cb16-4" tabindex="-1"></a><span class="co">#&gt;       sample  group</span></span>
<span id="cb16-5"><a href="#cb16-5" tabindex="-1"></a><span class="co">#&gt; 1   sample_1 group1</span></span>
<span id="cb16-6"><a href="#cb16-6" tabindex="-1"></a><span class="co">#&gt; 2   sample_2 group1</span></span>
<span id="cb16-7"><a href="#cb16-7" tabindex="-1"></a><span class="co">#&gt; 3   sample_3 group1</span></span>
<span id="cb16-8"><a href="#cb16-8" tabindex="-1"></a><span class="co">#&gt; 4   sample_4 group1</span></span>
<span id="cb16-9"><a href="#cb16-9" tabindex="-1"></a><span class="co">#&gt; 5   sample_5 group1</span></span>
<span id="cb16-10"><a href="#cb16-10" tabindex="-1"></a><span class="co">#&gt; 6   sample_6 group1</span></span>
<span id="cb16-11"><a href="#cb16-11" tabindex="-1"></a><span class="co">#&gt; 7   sample_7 group1</span></span>
<span id="cb16-12"><a href="#cb16-12" tabindex="-1"></a><span class="co">#&gt; 8   sample_8 group1</span></span>
<span id="cb16-13"><a href="#cb16-13" tabindex="-1"></a><span class="co">#&gt; 9   sample_9 group2</span></span>
<span id="cb16-14"><a href="#cb16-14" tabindex="-1"></a><span class="co">#&gt; 10 sample_10 group2</span></span>
<span id="cb16-15"><a href="#cb16-15" tabindex="-1"></a><span class="co">#&gt; 11 sample_11 group2</span></span>
<span id="cb16-16"><a href="#cb16-16" tabindex="-1"></a><span class="co">#&gt; 12 sample_12 group2</span></span>
<span id="cb16-17"><a href="#cb16-17" tabindex="-1"></a><span class="co">#&gt; 13 sample_13 group2</span></span>
<span id="cb16-18"><a href="#cb16-18" tabindex="-1"></a><span class="co">#&gt; 14 sample_14 group2</span></span>
<span id="cb16-19"><a href="#cb16-19" tabindex="-1"></a><span class="co">#&gt; 15 sample_15 group2</span></span>
<span id="cb16-20"><a href="#cb16-20" tabindex="-1"></a><span class="co">#&gt; 16 sample_16 group2</span></span>
<span id="cb16-21"><a href="#cb16-21" tabindex="-1"></a><span class="co">#&gt; 17 sample_17 group3</span></span>
<span id="cb16-22"><a href="#cb16-22" tabindex="-1"></a><span class="co">#&gt; 18 sample_18 group3</span></span>
<span id="cb16-23"><a href="#cb16-23" tabindex="-1"></a><span class="co">#&gt; 19 sample_19 group3</span></span>
<span id="cb16-24"><a href="#cb16-24" tabindex="-1"></a><span class="co">#&gt; 20 sample_20 group3</span></span>
<span id="cb16-25"><a href="#cb16-25" tabindex="-1"></a><span class="co">#&gt; 21 sample_21 group3</span></span>
<span id="cb16-26"><a href="#cb16-26" tabindex="-1"></a><span class="co">#&gt; 22 sample_22 group3</span></span>
<span id="cb16-27"><a href="#cb16-27" tabindex="-1"></a><span class="co">#&gt; 23 sample_23 group3</span></span>
<span id="cb16-28"><a href="#cb16-28" tabindex="-1"></a><span class="co">#&gt; 24 sample_24 group3</span></span></code></pre></div>
<p>Now we’ll apply the multigroup_main function to each of these tables,
again using lapply.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">lapply</span>(tables, multigroup_main, sample_info<span class="sc">$</span>group)</span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb17-3"><a href="#cb17-3" tabindex="-1"></a><span class="co">#&gt; Using one way anova </span></span>
<span id="cb17-4"><a href="#cb17-4" tabindex="-1"></a><span class="co">#&gt; anova 100% done</span></span>
<span id="cb17-5"><a href="#cb17-5" tabindex="-1"></a><span class="co">#&gt; Tukey test 100% done</span></span>
<span id="cb17-6"><a href="#cb17-6" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb17-7"><a href="#cb17-7" tabindex="-1"></a><span class="co">#&gt; Welch test 100% done</span></span>
<span id="cb17-8"><a href="#cb17-8" tabindex="-1"></a><span class="co">#&gt; DunnettT3 test 100% done</span></span>
<span id="cb17-9"><a href="#cb17-9" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb17-10"><a href="#cb17-10" tabindex="-1"></a><span class="co">#&gt; Kruskal-Wallis Test 100% done</span></span>
<span id="cb17-11"><a href="#cb17-11" tabindex="-1"></a><span class="co">#&gt; Dunn Test 100% done</span></span>
<span id="cb17-12"><a href="#cb17-12" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb17-13"><a href="#cb17-13" tabindex="-1"></a><span class="co">#&gt; sorting data 100% done...</span></span>
<span id="cb17-14"><a href="#cb17-14" tabindex="-1"></a><span class="co">#&gt; Using one way anova </span></span>
<span id="cb17-15"><a href="#cb17-15" tabindex="-1"></a><span class="co">#&gt; anova 100% done</span></span>
<span id="cb17-16"><a href="#cb17-16" tabindex="-1"></a><span class="co">#&gt; Tukey test 100% done</span></span>
<span id="cb17-17"><a href="#cb17-17" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb17-18"><a href="#cb17-18" tabindex="-1"></a><span class="co">#&gt; Welch test 100% done</span></span>
<span id="cb17-19"><a href="#cb17-19" tabindex="-1"></a><span class="co">#&gt; DunnettT3 test 100% done</span></span>
<span id="cb17-20"><a href="#cb17-20" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb17-21"><a href="#cb17-21" tabindex="-1"></a><span class="co">#&gt; Kruskal-Wallis Test 60% done...Kruskal-Wallis Test 100% done</span></span>
<span id="cb17-22"><a href="#cb17-22" tabindex="-1"></a><span class="co">#&gt; Dunn Test 60% done...Dunn Test 100% done</span></span>
<span id="cb17-23"><a href="#cb17-23" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb17-24"><a href="#cb17-24" tabindex="-1"></a><span class="co">#&gt; sorting data 100% done...</span></span>
<span id="cb17-25"><a href="#cb17-25" tabindex="-1"></a><span class="co">#&gt; Using one way anova </span></span>
<span id="cb17-26"><a href="#cb17-26" tabindex="-1"></a><span class="co">#&gt; anova 100% done</span></span>
<span id="cb17-27"><a href="#cb17-27" tabindex="-1"></a><span class="co">#&gt; Tukey test 100% done</span></span>
<span id="cb17-28"><a href="#cb17-28" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb17-29"><a href="#cb17-29" tabindex="-1"></a><span class="co">#&gt; Kruskal-Wallis Test 71% done...Kruskal-Wallis Test 100% done</span></span>
<span id="cb17-30"><a href="#cb17-30" tabindex="-1"></a><span class="co">#&gt; Dunn Test 71% done...Dunn Test 100% done</span></span>
<span id="cb17-31"><a href="#cb17-31" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb17-32"><a href="#cb17-32" tabindex="-1"></a><span class="co">#&gt; sorting data 100% done...</span></span>
<span id="cb17-33"><a href="#cb17-33" tabindex="-1"></a><span class="co">#&gt; Using one way anova </span></span>
<span id="cb17-34"><a href="#cb17-34" tabindex="-1"></a><span class="co">#&gt; anova 100% done</span></span>
<span id="cb17-35"><a href="#cb17-35" tabindex="-1"></a><span class="co">#&gt; Tukey test 100% done</span></span>
<span id="cb17-36"><a href="#cb17-36" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb17-37"><a href="#cb17-37" tabindex="-1"></a><span class="co">#&gt; Welch test 100% done</span></span>
<span id="cb17-38"><a href="#cb17-38" tabindex="-1"></a><span class="co">#&gt; DunnettT3 test 100% done</span></span>
<span id="cb17-39"><a href="#cb17-39" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb17-40"><a href="#cb17-40" tabindex="-1"></a><span class="co">#&gt; Kruskal-Wallis Test 60% done...Kruskal-Wallis Test 100% done</span></span>
<span id="cb17-41"><a href="#cb17-41" tabindex="-1"></a><span class="co">#&gt; Dunn Test 60% done...Dunn Test 100% done</span></span>
<span id="cb17-42"><a href="#cb17-42" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb17-43"><a href="#cb17-43" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb17-44"><a href="#cb17-44" tabindex="-1"></a><span class="co">#&gt; Using one way anova </span></span>
<span id="cb17-45"><a href="#cb17-45" tabindex="-1"></a><span class="co">#&gt; anova 100% done</span></span>
<span id="cb17-46"><a href="#cb17-46" tabindex="-1"></a><span class="co">#&gt; Tukey test 100% done</span></span>
<span id="cb17-47"><a href="#cb17-47" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb17-48"><a href="#cb17-48" tabindex="-1"></a><span class="co">#&gt; Welch test 100% done</span></span>
<span id="cb17-49"><a href="#cb17-49" tabindex="-1"></a><span class="co">#&gt; DunnettT3 test 100% done</span></span>
<span id="cb17-50"><a href="#cb17-50" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb17-51"><a href="#cb17-51" tabindex="-1"></a><span class="co">#&gt; Kruskal-Wallis Test 100% done</span></span>
<span id="cb17-52"><a href="#cb17-52" tabindex="-1"></a><span class="co">#&gt; Dunn Test 100% done</span></span>
<span id="cb17-53"><a href="#cb17-53" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb17-54"><a href="#cb17-54" tabindex="-1"></a><span class="co">#&gt; sorting data 100% done...</span></span>
<span id="cb17-55"><a href="#cb17-55" tabindex="-1"></a><span class="co">#&gt; Using one way anova </span></span>
<span id="cb17-56"><a href="#cb17-56" tabindex="-1"></a><span class="co">#&gt; anova 100% done</span></span>
<span id="cb17-57"><a href="#cb17-57" tabindex="-1"></a><span class="co">#&gt; Tukey test 100% done</span></span>
<span id="cb17-58"><a href="#cb17-58" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb17-59"><a href="#cb17-59" tabindex="-1"></a><span class="co">#&gt; Welch test 100% done</span></span>
<span id="cb17-60"><a href="#cb17-60" tabindex="-1"></a><span class="co">#&gt; DunnettT3 test 100% done</span></span>
<span id="cb17-61"><a href="#cb17-61" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb17-62"><a href="#cb17-62" tabindex="-1"></a><span class="co">#&gt; Kruskal-Wallis Test 63% done...Kruskal-Wallis Test 100% done</span></span>
<span id="cb17-63"><a href="#cb17-63" tabindex="-1"></a><span class="co">#&gt; Dunn Test 63% done...Dunn Test 100% done</span></span>
<span id="cb17-64"><a href="#cb17-64" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span></code></pre></div>
<p>We know from before that multigroup_main returns a list, and we just
applied this function to a list of data.frames. So what is res? It’s a
list of lists.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="co">#res has 6 elements, one per table we analyzed</span></span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a><span class="fu">length</span>(res)</span>
<span id="cb18-3"><a href="#cb18-3" tabindex="-1"></a><span class="co">#&gt; [1] 6</span></span>
<span id="cb18-4"><a href="#cb18-4" tabindex="-1"></a><span class="co">#each element of res has 7 elements</span></span>
<span id="cb18-5"><a href="#cb18-5" tabindex="-1"></a><span class="fu">length</span>(res[[<span class="dv">1</span>]])</span>
<span id="cb18-6"><a href="#cb18-6" tabindex="-1"></a><span class="co">#&gt; [1] 7</span></span>
<span id="cb18-7"><a href="#cb18-7" tabindex="-1"></a><span class="fu">names</span>(res[[<span class="dv">1</span>]])</span>
<span id="cb18-8"><a href="#cb18-8" tabindex="-1"></a><span class="co">#&gt; [1] &quot;anova_results&quot; &quot;welch_results&quot; &quot;kw_results&quot;    &quot;foldchanges&quot;  </span></span>
<span id="cb18-9"><a href="#cb18-9" tabindex="-1"></a><span class="co">#&gt; [5] &quot;ne&quot;            &quot;nu&quot;            &quot;nn&quot;</span></span></code></pre></div>
<p>Each element of res is the same as it was when we were only analyzing
one table. The only difference is now we’re working with several of them
at once.</p>
<p>We’ll look at the results of the first one just to orient ourselves,
and we’re only going to look at the first three elements, the anova,
welch, and kruskal-wallis results.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a><span class="co">#get the first element of res (which is a list of results of the first analysis). This will be the class abundances since it is first alphabetically</span></span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a>x <span class="ot">&lt;-</span> res[[<span class="dv">1</span>]]</span>
<span id="cb19-3"><a href="#cb19-3" tabindex="-1"></a><span class="fu">lapply</span>(x[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>], head)</span>
<span id="cb19-4"><a href="#cb19-4" tabindex="-1"></a><span class="co">#&gt; $anova_results</span></span>
<span id="cb19-5"><a href="#cb19-5" tabindex="-1"></a><span class="co">#&gt;                           Pval Pval_group1_vs_group2</span></span>
<span id="cb19-6"><a href="#cb19-6" tabindex="-1"></a><span class="co">#&gt; Betaproteobacteria  0.20000097             0.8263342</span></span>
<span id="cb19-7"><a href="#cb19-7" tabindex="-1"></a><span class="co">#&gt; Coriobacteriia      0.06770511             0.5440510</span></span>
<span id="cb19-8"><a href="#cb19-8" tabindex="-1"></a><span class="co">#&gt; Negativicutes       0.04907064             0.6193168</span></span>
<span id="cb19-9"><a href="#cb19-9" tabindex="-1"></a><span class="co">#&gt; Tissierellia        0.15297673             0.1656226</span></span>
<span id="cb19-10"><a href="#cb19-10" tabindex="-1"></a><span class="co">#&gt; Alphaproteobacteria 0.08962253             0.9493171</span></span>
<span id="cb19-11"><a href="#cb19-11" tabindex="-1"></a><span class="co">#&gt; Flavobacteriia      0.16490962             0.5901316</span></span>
<span id="cb19-12"><a href="#cb19-12" tabindex="-1"></a><span class="co">#&gt;                     log2FoldChange_group1_vs_group2 Pval_group1_vs_group3</span></span>
<span id="cb19-13"><a href="#cb19-13" tabindex="-1"></a><span class="co">#&gt; Betaproteobacteria                       -0.2803858             0.4456629</span></span>
<span id="cb19-14"><a href="#cb19-14" tabindex="-1"></a><span class="co">#&gt; Coriobacteriia                           -0.5572681             0.3576415</span></span>
<span id="cb19-15"><a href="#cb19-15" tabindex="-1"></a><span class="co">#&gt; Negativicutes                             0.6147098             0.2415234</span></span>
<span id="cb19-16"><a href="#cb19-16" tabindex="-1"></a><span class="co">#&gt; Tissierellia                              1.2537566             0.9467810</span></span>
<span id="cb19-17"><a href="#cb19-17" tabindex="-1"></a><span class="co">#&gt; Alphaproteobacteria                       0.2630344             0.1795866</span></span>
<span id="cb19-18"><a href="#cb19-18" tabindex="-1"></a><span class="co">#&gt; Flavobacteriia                           -0.8744691             0.1411272</span></span>
<span id="cb19-19"><a href="#cb19-19" tabindex="-1"></a><span class="co">#&gt;                     log2FoldChange_group1_vs_group3 Pval_group2_vs_group3</span></span>
<span id="cb19-20"><a href="#cb19-20" tabindex="-1"></a><span class="co">#&gt; Betaproteobacteria                        0.8590642            0.18512474</span></span>
<span id="cb19-21"><a href="#cb19-21" tabindex="-1"></a><span class="co">#&gt; Coriobacteriia                            1.3940111            0.05543407</span></span>
<span id="cb19-22"><a href="#cb19-22" tabindex="-1"></a><span class="co">#&gt; Negativicutes                            -0.6890709            0.04158533</span></span>
<span id="cb19-23"><a href="#cb19-23" tabindex="-1"></a><span class="co">#&gt; Tissierellia                              0.1468414            0.27729400</span></span>
<span id="cb19-24"><a href="#cb19-24" tabindex="-1"></a><span class="co">#&gt; Alphaproteobacteria                      -1.0000000            0.10326409</span></span>
<span id="cb19-25"><a href="#cb19-25" tabindex="-1"></a><span class="co">#&gt; Flavobacteriia                           -1.4150375            0.59013161</span></span>
<span id="cb19-26"><a href="#cb19-26" tabindex="-1"></a><span class="co">#&gt;                     log2FoldChange_group2_vs_group3 group1_mean group2_mean</span></span>
<span id="cb19-27"><a href="#cb19-27" tabindex="-1"></a><span class="co">#&gt; Betaproteobacteria                        1.1394500     229.000     278.125</span></span>
<span id="cb19-28"><a href="#cb19-28" tabindex="-1"></a><span class="co">#&gt; Coriobacteriia                            1.9512792     173.125     254.750</span></span>
<span id="cb19-29"><a href="#cb19-29" tabindex="-1"></a><span class="co">#&gt; Negativicutes                            -1.3037807       6.125       4.000</span></span>
<span id="cb19-30"><a href="#cb19-30" tabindex="-1"></a><span class="co">#&gt; Tissierellia                             -1.1069152       3.875       1.625</span></span>
<span id="cb19-31"><a href="#cb19-31" tabindex="-1"></a><span class="co">#&gt; Alphaproteobacteria                      -1.2630344       1.500       1.250</span></span>
<span id="cb19-32"><a href="#cb19-32" tabindex="-1"></a><span class="co">#&gt; Flavobacteriia                           -0.5405684       0.750       1.375</span></span>
<span id="cb19-33"><a href="#cb19-33" tabindex="-1"></a><span class="co">#&gt;                     group3_mean       test</span></span>
<span id="cb19-34"><a href="#cb19-34" tabindex="-1"></a><span class="co">#&gt; Betaproteobacteria      126.250 tukey-test</span></span>
<span id="cb19-35"><a href="#cb19-35" tabindex="-1"></a><span class="co">#&gt; Coriobacteriia           65.875 tukey-test</span></span>
<span id="cb19-36"><a href="#cb19-36" tabindex="-1"></a><span class="co">#&gt; Negativicutes             9.875 tukey-test</span></span>
<span id="cb19-37"><a href="#cb19-37" tabindex="-1"></a><span class="co">#&gt; Tissierellia              3.500 tukey-test</span></span>
<span id="cb19-38"><a href="#cb19-38" tabindex="-1"></a><span class="co">#&gt; Alphaproteobacteria       3.000 tukey-test</span></span>
<span id="cb19-39"><a href="#cb19-39" tabindex="-1"></a><span class="co">#&gt; Flavobacteriia            2.000 tukey-test</span></span>
<span id="cb19-40"><a href="#cb19-40" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb19-41"><a href="#cb19-41" tabindex="-1"></a><span class="co">#&gt; $welch_results</span></span>
<span id="cb19-42"><a href="#cb19-42" tabindex="-1"></a><span class="co">#&gt;                     Pval Pval_group1_vs_group2 log2FoldChange_group1_vs_group2</span></span>
<span id="cb19-43"><a href="#cb19-43" tabindex="-1"></a><span class="co">#&gt; Bacteroidia   0.03093208             0.2372938                       0.5706097</span></span>
<span id="cb19-44"><a href="#cb19-44" tabindex="-1"></a><span class="co">#&gt; Chitinophagia 0.04254664             0.6030114                       1.2223924</span></span>
<span id="cb19-45"><a href="#cb19-45" tabindex="-1"></a><span class="co">#&gt;               Pval_group1_vs_group3 log2FoldChange_group1_vs_group3</span></span>
<span id="cb19-46"><a href="#cb19-46" tabindex="-1"></a><span class="co">#&gt; Bacteroidia               0.2150544                      -0.6241665</span></span>
<span id="cb19-47"><a href="#cb19-47" tabindex="-1"></a><span class="co">#&gt; Chitinophagia             0.2606331                      -1.2801079</span></span>
<span id="cb19-48"><a href="#cb19-48" tabindex="-1"></a><span class="co">#&gt;               Pval_group2_vs_group3 log2FoldChange_group2_vs_group3 group1_mean</span></span>
<span id="cb19-49"><a href="#cb19-49" tabindex="-1"></a><span class="co">#&gt; Bacteroidia              0.03416268                       -1.194776     706.375</span></span>
<span id="cb19-50"><a href="#cb19-50" tabindex="-1"></a><span class="co">#&gt; Chitinophagia            0.05656468                       -2.502500       0.875</span></span>
<span id="cb19-51"><a href="#cb19-51" tabindex="-1"></a><span class="co">#&gt;               group2_mean group3_mean         test</span></span>
<span id="cb19-52"><a href="#cb19-52" tabindex="-1"></a><span class="co">#&gt; Bacteroidia       475.625    1088.750 dunnett-test</span></span>
<span id="cb19-53"><a href="#cb19-53" tabindex="-1"></a><span class="co">#&gt; Chitinophagia       0.375       2.125 dunnett-test</span></span>
<span id="cb19-54"><a href="#cb19-54" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb19-55"><a href="#cb19-55" tabindex="-1"></a><span class="co">#&gt; $kw_results</span></span>
<span id="cb19-56"><a href="#cb19-56" tabindex="-1"></a><span class="co">#&gt;                           Pval Pval_group1_vs_group2</span></span>
<span id="cb19-57"><a href="#cb19-57" tabindex="-1"></a><span class="co">#&gt; Clostridia          0.81668648                     1</span></span>
<span id="cb19-58"><a href="#cb19-58" tabindex="-1"></a><span class="co">#&gt; Erysipelotrichia    0.65079423                     1</span></span>
<span id="cb19-59"><a href="#cb19-59" tabindex="-1"></a><span class="co">#&gt; unclassified        0.73334806                     1</span></span>
<span id="cb19-60"><a href="#cb19-60" tabindex="-1"></a><span class="co">#&gt; Verrucomicrobiae    0.12045536                     1</span></span>
<span id="cb19-61"><a href="#cb19-61" tabindex="-1"></a><span class="co">#&gt; Bacilli             0.08206044                     1</span></span>
<span id="cb19-62"><a href="#cb19-62" tabindex="-1"></a><span class="co">#&gt; Gammaproteobacteria 0.79132952                     1</span></span>
<span id="cb19-63"><a href="#cb19-63" tabindex="-1"></a><span class="co">#&gt;                     log2FoldChange_group1_vs_group2 Pval_group1_vs_group3</span></span>
<span id="cb19-64"><a href="#cb19-64" tabindex="-1"></a><span class="co">#&gt; Clostridia                               0.06548115            1.00000000</span></span>
<span id="cb19-65"><a href="#cb19-65" tabindex="-1"></a><span class="co">#&gt; Erysipelotrichia                         0.45625701            1.00000000</span></span>
<span id="cb19-66"><a href="#cb19-66" tabindex="-1"></a><span class="co">#&gt; unclassified                            -0.28237212            1.00000000</span></span>
<span id="cb19-67"><a href="#cb19-67" tabindex="-1"></a><span class="co">#&gt; Verrucomicrobiae                        -0.47375524            0.19709373</span></span>
<span id="cb19-68"><a href="#cb19-68" tabindex="-1"></a><span class="co">#&gt; Bacilli                                 -2.15324626            0.08117608</span></span>
<span id="cb19-69"><a href="#cb19-69" tabindex="-1"></a><span class="co">#&gt; Gammaproteobacteria                     -0.46247849            1.00000000</span></span>
<span id="cb19-70"><a href="#cb19-70" tabindex="-1"></a><span class="co">#&gt;                     log2FoldChange_group1_vs_group3 Pval_group2_vs_group3</span></span>
<span id="cb19-71"><a href="#cb19-71" tabindex="-1"></a><span class="co">#&gt; Clostridia                              0.007316499             1.0000000</span></span>
<span id="cb19-72"><a href="#cb19-72" tabindex="-1"></a><span class="co">#&gt; Erysipelotrichia                       -0.309751171             1.0000000</span></span>
<span id="cb19-73"><a href="#cb19-73" tabindex="-1"></a><span class="co">#&gt; unclassified                           -0.561999137             1.0000000</span></span>
<span id="cb19-74"><a href="#cb19-74" tabindex="-1"></a><span class="co">#&gt; Verrucomicrobiae                       -0.652276613             0.2581564</span></span>
<span id="cb19-75"><a href="#cb19-75" tabindex="-1"></a><span class="co">#&gt; Bacilli                                -2.106153552             0.4871113</span></span>
<span id="cb19-76"><a href="#cb19-76" tabindex="-1"></a><span class="co">#&gt; Gammaproteobacteria                    -0.173648087             1.0000000</span></span>
<span id="cb19-77"><a href="#cb19-77" tabindex="-1"></a><span class="co">#&gt;                     log2FoldChange_group2_vs_group3 group1_mean group2_mean</span></span>
<span id="cb19-78"><a href="#cb19-78" tabindex="-1"></a><span class="co">#&gt; Clostridia                              -0.05816465    1408.500    1346.000</span></span>
<span id="cb19-79"><a href="#cb19-79" tabindex="-1"></a><span class="co">#&gt; Erysipelotrichia                        -0.76600818     276.625     201.625</span></span>
<span id="cb19-80"><a href="#cb19-80" tabindex="-1"></a><span class="co">#&gt; unclassified                            -0.27962702     145.125     176.500</span></span>
<span id="cb19-81"><a href="#cb19-81" tabindex="-1"></a><span class="co">#&gt; Verrucomicrobiae                        -0.17852137      82.000     113.875</span></span>
<span id="cb19-82"><a href="#cb19-82" tabindex="-1"></a><span class="co">#&gt; Bacilli                                  0.04709271      25.375     112.875</span></span>
<span id="cb19-83"><a href="#cb19-83" tabindex="-1"></a><span class="co">#&gt; Gammaproteobacteria                      0.28883041      21.500      29.625</span></span>
<span id="cb19-84"><a href="#cb19-84" tabindex="-1"></a><span class="co">#&gt;                     group3_mean      test</span></span>
<span id="cb19-85"><a href="#cb19-85" tabindex="-1"></a><span class="co">#&gt; Clostridia             1401.375 dunn-test</span></span>
<span id="cb19-86"><a href="#cb19-86" tabindex="-1"></a><span class="co">#&gt; Erysipelotrichia        342.875 dunn-test</span></span>
<span id="cb19-87"><a href="#cb19-87" tabindex="-1"></a><span class="co">#&gt; unclassified            214.250 dunn-test</span></span>
<span id="cb19-88"><a href="#cb19-88" tabindex="-1"></a><span class="co">#&gt; Verrucomicrobiae        128.875 dunn-test</span></span>
<span id="cb19-89"><a href="#cb19-89" tabindex="-1"></a><span class="co">#&gt; Bacilli                 109.250 dunn-test</span></span>
<span id="cb19-90"><a href="#cb19-90" tabindex="-1"></a><span class="co">#&gt; Gammaproteobacteria      24.250 dunn-test</span></span></code></pre></div>
<p>Since all the results of this analysis were tested in the same way,
and all of the results from each taxonomic level have the same columns,
we can collate each one into a single data.frame.</p>
<p>What we’ll do is for each list of results in res, we’ll rbind (row
bind) the first three elements, save this collated table to a new list
and then give it a name.</p>
<p>For the names, we’ll use the result of the list.files function above,
but we’ll trim off the .csv extension first.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a><span class="co">#gsub .csv with nothing in list a</span></span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a>name_list <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;.csv&quot;</span>, <span class="st">&quot;&quot;</span>, a)</span>
<span id="cb20-3"><a href="#cb20-3" tabindex="-1"></a>name_list</span>
<span id="cb20-4"><a href="#cb20-4" tabindex="-1"></a><span class="co">#&gt; [1] &quot;class_abundance&quot;   &quot;family_abundance&quot;  &quot;genus_abundance&quot;  </span></span>
<span id="cb20-5"><a href="#cb20-5" tabindex="-1"></a><span class="co">#&gt; [4] &quot;order_abundance&quot;   &quot;phylum_abundance&quot;  &quot;species_abundance&quot;</span></span></code></pre></div>
<p>Now we’ll collate and save the tables</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a><span class="co">#define an empty list for our new tables</span></span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a>collated_tables <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb21-3"><a href="#cb21-3" tabindex="-1"></a><span class="co">#use a for loop to collate and save our tables</span></span>
<span id="cb21-4"><a href="#cb21-4" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(res)) {</span>
<span id="cb21-5"><a href="#cb21-5" tabindex="-1"></a>  <span class="co">#the current result were processing</span></span>
<span id="cb21-6"><a href="#cb21-6" tabindex="-1"></a>  temp <span class="ot">&lt;-</span> res[[i]]</span>
<span id="cb21-7"><a href="#cb21-7" tabindex="-1"></a>  <span class="co">#get just the p values and fold change results </span></span>
<span id="cb21-8"><a href="#cb21-8" tabindex="-1"></a>  first_three <span class="ot">&lt;-</span> temp[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span>
<span id="cb21-9"><a href="#cb21-9" tabindex="-1"></a>  <span class="co">#rbind into one new table</span></span>
<span id="cb21-10"><a href="#cb21-10" tabindex="-1"></a>  new_table <span class="ot">&lt;-</span> <span class="fu">rbind</span>(first_three[[<span class="dv">1</span>]], first_three[[<span class="dv">2</span>]], first_three[[<span class="dv">3</span>]])</span>
<span id="cb21-11"><a href="#cb21-11" tabindex="-1"></a>  <span class="co">#save the new table to the collated_tables list</span></span>
<span id="cb21-12"><a href="#cb21-12" tabindex="-1"></a>  collated_tables[[i]] <span class="ot">&lt;-</span> new_table</span>
<span id="cb21-13"><a href="#cb21-13" tabindex="-1"></a>}</span>
<span id="cb21-14"><a href="#cb21-14" tabindex="-1"></a><span class="co">#set the names of collated_tables to be the names from name_list</span></span>
<span id="cb21-15"><a href="#cb21-15" tabindex="-1"></a><span class="fu">names</span>(collated_tables) <span class="ot">&lt;-</span> name_list</span>
<span id="cb21-16"><a href="#cb21-16" tabindex="-1"></a><span class="co">#look at some of the results</span></span>
<span id="cb21-17"><a href="#cb21-17" tabindex="-1"></a><span class="fu">lapply</span>(collated_tables, head, <span class="dv">3</span>)</span>
<span id="cb21-18"><a href="#cb21-18" tabindex="-1"></a><span class="co">#&gt; $class_abundance</span></span>
<span id="cb21-19"><a href="#cb21-19" tabindex="-1"></a><span class="co">#&gt;                          Pval Pval_group1_vs_group2</span></span>
<span id="cb21-20"><a href="#cb21-20" tabindex="-1"></a><span class="co">#&gt; Betaproteobacteria 0.20000097             0.8263342</span></span>
<span id="cb21-21"><a href="#cb21-21" tabindex="-1"></a><span class="co">#&gt; Coriobacteriia     0.06770511             0.5440510</span></span>
<span id="cb21-22"><a href="#cb21-22" tabindex="-1"></a><span class="co">#&gt; Negativicutes      0.04907064             0.6193168</span></span>
<span id="cb21-23"><a href="#cb21-23" tabindex="-1"></a><span class="co">#&gt;                    log2FoldChange_group1_vs_group2 Pval_group1_vs_group3</span></span>
<span id="cb21-24"><a href="#cb21-24" tabindex="-1"></a><span class="co">#&gt; Betaproteobacteria                      -0.2803858             0.4456629</span></span>
<span id="cb21-25"><a href="#cb21-25" tabindex="-1"></a><span class="co">#&gt; Coriobacteriia                          -0.5572681             0.3576415</span></span>
<span id="cb21-26"><a href="#cb21-26" tabindex="-1"></a><span class="co">#&gt; Negativicutes                            0.6147098             0.2415234</span></span>
<span id="cb21-27"><a href="#cb21-27" tabindex="-1"></a><span class="co">#&gt;                    log2FoldChange_group1_vs_group3 Pval_group2_vs_group3</span></span>
<span id="cb21-28"><a href="#cb21-28" tabindex="-1"></a><span class="co">#&gt; Betaproteobacteria                       0.8590642            0.18512474</span></span>
<span id="cb21-29"><a href="#cb21-29" tabindex="-1"></a><span class="co">#&gt; Coriobacteriia                           1.3940111            0.05543407</span></span>
<span id="cb21-30"><a href="#cb21-30" tabindex="-1"></a><span class="co">#&gt; Negativicutes                           -0.6890709            0.04158533</span></span>
<span id="cb21-31"><a href="#cb21-31" tabindex="-1"></a><span class="co">#&gt;                    log2FoldChange_group2_vs_group3 group1_mean group2_mean</span></span>
<span id="cb21-32"><a href="#cb21-32" tabindex="-1"></a><span class="co">#&gt; Betaproteobacteria                        1.139450     229.000     278.125</span></span>
<span id="cb21-33"><a href="#cb21-33" tabindex="-1"></a><span class="co">#&gt; Coriobacteriia                            1.951279     173.125     254.750</span></span>
<span id="cb21-34"><a href="#cb21-34" tabindex="-1"></a><span class="co">#&gt; Negativicutes                            -1.303781       6.125       4.000</span></span>
<span id="cb21-35"><a href="#cb21-35" tabindex="-1"></a><span class="co">#&gt;                    group3_mean       test</span></span>
<span id="cb21-36"><a href="#cb21-36" tabindex="-1"></a><span class="co">#&gt; Betaproteobacteria     126.250 tukey-test</span></span>
<span id="cb21-37"><a href="#cb21-37" tabindex="-1"></a><span class="co">#&gt; Coriobacteriia          65.875 tukey-test</span></span>
<span id="cb21-38"><a href="#cb21-38" tabindex="-1"></a><span class="co">#&gt; Negativicutes            9.875 tukey-test</span></span>
<span id="cb21-39"><a href="#cb21-39" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb21-40"><a href="#cb21-40" tabindex="-1"></a><span class="co">#&gt; $family_abundance</span></span>
<span id="cb21-41"><a href="#cb21-41" tabindex="-1"></a><span class="co">#&gt;                       Pval Pval_group1_vs_group2</span></span>
<span id="cb21-42"><a href="#cb21-42" tabindex="-1"></a><span class="co">#&gt; Sutterellaceae 0.197457096             0.8104024</span></span>
<span id="cb21-43"><a href="#cb21-43" tabindex="-1"></a><span class="co">#&gt; Atopobiaceae   0.066157338             0.5388124</span></span>
<span id="cb21-44"><a href="#cb21-44" tabindex="-1"></a><span class="co">#&gt; Prevotellaceae 0.001577777             0.6487139</span></span>
<span id="cb21-45"><a href="#cb21-45" tabindex="-1"></a><span class="co">#&gt;                log2FoldChange_group1_vs_group2 Pval_group1_vs_group3</span></span>
<span id="cb21-46"><a href="#cb21-46" tabindex="-1"></a><span class="co">#&gt; Sutterellaceae                      -0.2966753            0.45486902</span></span>
<span id="cb21-47"><a href="#cb21-47" tabindex="-1"></a><span class="co">#&gt; Atopobiaceae                        -0.5663191            0.35585660</span></span>
<span id="cb21-48"><a href="#cb21-48" tabindex="-1"></a><span class="co">#&gt; Prevotellaceae                       0.7670228            0.01360419</span></span>
<span id="cb21-49"><a href="#cb21-49" tabindex="-1"></a><span class="co">#&gt;                log2FoldChange_group1_vs_group3 Pval_group2_vs_group3</span></span>
<span id="cb21-50"><a href="#cb21-50" tabindex="-1"></a><span class="co">#&gt; Sutterellaceae                       0.8589077            0.18096687</span></span>
<span id="cb21-51"><a href="#cb21-51" tabindex="-1"></a><span class="co">#&gt; Atopobiaceae                         1.4272001            0.05406228</span></span>
<span id="cb21-52"><a href="#cb21-52" tabindex="-1"></a><span class="co">#&gt; Prevotellaceae                      -1.2857835            0.00171722</span></span>
<span id="cb21-53"><a href="#cb21-53" tabindex="-1"></a><span class="co">#&gt;                log2FoldChange_group2_vs_group3 group1_mean group2_mean</span></span>
<span id="cb21-54"><a href="#cb21-54" tabindex="-1"></a><span class="co">#&gt; Sutterellaceae                        1.155583      219.00     269.000</span></span>
<span id="cb21-55"><a href="#cb21-55" tabindex="-1"></a><span class="co">#&gt; Atopobiaceae                          1.993519      168.75     249.875</span></span>
<span id="cb21-56"><a href="#cb21-56" tabindex="-1"></a><span class="co">#&gt; Prevotellaceae                       -2.052806       24.25      14.250</span></span>
<span id="cb21-57"><a href="#cb21-57" tabindex="-1"></a><span class="co">#&gt;                group3_mean       test</span></span>
<span id="cb21-58"><a href="#cb21-58" tabindex="-1"></a><span class="co">#&gt; Sutterellaceae     120.750 tukey-test</span></span>
<span id="cb21-59"><a href="#cb21-59" tabindex="-1"></a><span class="co">#&gt; Atopobiaceae        62.750 tukey-test</span></span>
<span id="cb21-60"><a href="#cb21-60" tabindex="-1"></a><span class="co">#&gt; Prevotellaceae      59.125 tukey-test</span></span>
<span id="cb21-61"><a href="#cb21-61" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb21-62"><a href="#cb21-62" tabindex="-1"></a><span class="co">#&gt; $genus_abundance</span></span>
<span id="cb21-63"><a href="#cb21-63" tabindex="-1"></a><span class="co">#&gt;                          Pval Pval_group1_vs_group2</span></span>
<span id="cb21-64"><a href="#cb21-64" tabindex="-1"></a><span class="co">#&gt; Parasutterella     0.19869563             0.8113967</span></span>
<span id="cb21-65"><a href="#cb21-65" tabindex="-1"></a><span class="co">#&gt; Muricaecibacterium 0.06812654             0.5464864</span></span>
<span id="cb21-66"><a href="#cb21-66" tabindex="-1"></a><span class="co">#&gt; Anaerotruncus      0.51580712             0.8242216</span></span>
<span id="cb21-67"><a href="#cb21-67" tabindex="-1"></a><span class="co">#&gt;                    log2FoldChange_group1_vs_group2 Pval_group1_vs_group3</span></span>
<span id="cb21-68"><a href="#cb21-68" tabindex="-1"></a><span class="co">#&gt; Parasutterella                          -0.2961576             0.4560574</span></span>
<span id="cb21-69"><a href="#cb21-69" tabindex="-1"></a><span class="co">#&gt; Muricaecibacterium                      -0.5635121             0.3573489</span></span>
<span id="cb21-70"><a href="#cb21-70" tabindex="-1"></a><span class="co">#&gt; Anaerotruncus                            0.3156143             0.8353710</span></span>
<span id="cb21-71"><a href="#cb21-71" tabindex="-1"></a><span class="co">#&gt;                    log2FoldChange_group1_vs_group3 Pval_group2_vs_group3</span></span>
<span id="cb21-72"><a href="#cb21-72" tabindex="-1"></a><span class="co">#&gt; Parasutterella                           0.8580840            0.18218500</span></span>
<span id="cb21-73"><a href="#cb21-73" tabindex="-1"></a><span class="co">#&gt; Muricaecibacterium                       1.4364879            0.05582506</span></span>
<span id="cb21-74"><a href="#cb21-74" tabindex="-1"></a><span class="co">#&gt; Anaerotruncus                           -0.2503234            0.48405009</span></span>
<span id="cb21-75"><a href="#cb21-75" tabindex="-1"></a><span class="co">#&gt;                    log2FoldChange_group2_vs_group3 group1_mean group2_mean</span></span>
<span id="cb21-76"><a href="#cb21-76" tabindex="-1"></a><span class="co">#&gt; Parasutterella                           1.1542416     218.875     268.750</span></span>
<span id="cb21-77"><a href="#cb21-77" tabindex="-1"></a><span class="co">#&gt; Muricaecibacterium                       2.0000000     163.750     242.000</span></span>
<span id="cb21-78"><a href="#cb21-78" tabindex="-1"></a><span class="co">#&gt; Anaerotruncus                           -0.5659377      35.625      28.625</span></span>
<span id="cb21-79"><a href="#cb21-79" tabindex="-1"></a><span class="co">#&gt;                    group3_mean       test</span></span>
<span id="cb21-80"><a href="#cb21-80" tabindex="-1"></a><span class="co">#&gt; Parasutterella         120.750 tukey-test</span></span>
<span id="cb21-81"><a href="#cb21-81" tabindex="-1"></a><span class="co">#&gt; Muricaecibacterium      60.500 tukey-test</span></span>
<span id="cb21-82"><a href="#cb21-82" tabindex="-1"></a><span class="co">#&gt; Anaerotruncus           42.375 tukey-test</span></span>
<span id="cb21-83"><a href="#cb21-83" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb21-84"><a href="#cb21-84" tabindex="-1"></a><span class="co">#&gt; $order_abundance</span></span>
<span id="cb21-85"><a href="#cb21-85" tabindex="-1"></a><span class="co">#&gt;                        Pval Pval_group1_vs_group2</span></span>
<span id="cb21-86"><a href="#cb21-86" tabindex="-1"></a><span class="co">#&gt; Burkholderiales  0.19886795             0.8240061</span></span>
<span id="cb21-87"><a href="#cb21-87" tabindex="-1"></a><span class="co">#&gt; Coriobacteriales 0.06626659             0.5465790</span></span>
<span id="cb21-88"><a href="#cb21-88" tabindex="-1"></a><span class="co">#&gt; Bacillales       0.04799237             0.1124440</span></span>
<span id="cb21-89"><a href="#cb21-89" tabindex="-1"></a><span class="co">#&gt;                  log2FoldChange_group1_vs_group2 Pval_group1_vs_group3</span></span>
<span id="cb21-90"><a href="#cb21-90" tabindex="-1"></a><span class="co">#&gt; Burkholderiales                       -0.2828907             0.4457946</span></span>
<span id="cb21-91"><a href="#cb21-91" tabindex="-1"></a><span class="co">#&gt; Coriobacteriales                      -0.5561823             0.3504407</span></span>
<span id="cb21-92"><a href="#cb21-92" tabindex="-1"></a><span class="co">#&gt; Bacillales                             1.1020982             0.9373723</span></span>
<span id="cb21-93"><a href="#cb21-93" tabindex="-1"></a><span class="co">#&gt;                  log2FoldChange_group1_vs_group3 Pval_group2_vs_group3</span></span>
<span id="cb21-94"><a href="#cb21-94" tabindex="-1"></a><span class="co">#&gt; Burkholderiales                        0.8616358            0.18382079</span></span>
<span id="cb21-95"><a href="#cb21-95" tabindex="-1"></a><span class="co">#&gt; Coriobacteriales                       1.4224441            0.05427840</span></span>
<span id="cb21-96"><a href="#cb21-96" tabindex="-1"></a><span class="co">#&gt; Bacillales                            -0.1202942            0.05767428</span></span>
<span id="cb21-97"><a href="#cb21-97" tabindex="-1"></a><span class="co">#&gt;                  log2FoldChange_group2_vs_group3 group1_mean group2_mean</span></span>
<span id="cb21-98"><a href="#cb21-98" tabindex="-1"></a><span class="co">#&gt; Burkholderiales                         1.144526     228.500     278.000</span></span>
<span id="cb21-99"><a href="#cb21-99" tabindex="-1"></a><span class="co">#&gt; Coriobacteriales                        1.978626     170.875     251.250</span></span>
<span id="cb21-100"><a href="#cb21-100" tabindex="-1"></a><span class="co">#&gt; Bacillales                             -1.222392      20.125       9.375</span></span>
<span id="cb21-101"><a href="#cb21-101" tabindex="-1"></a><span class="co">#&gt;                  group3_mean       test</span></span>
<span id="cb21-102"><a href="#cb21-102" tabindex="-1"></a><span class="co">#&gt; Burkholderiales      125.750 tukey-test</span></span>
<span id="cb21-103"><a href="#cb21-103" tabindex="-1"></a><span class="co">#&gt; Coriobacteriales      63.750 tukey-test</span></span>
<span id="cb21-104"><a href="#cb21-104" tabindex="-1"></a><span class="co">#&gt; Bacillales            21.875 tukey-test</span></span>
<span id="cb21-105"><a href="#cb21-105" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb21-106"><a href="#cb21-106" tabindex="-1"></a><span class="co">#&gt; $phylum_abundance</span></span>
<span id="cb21-107"><a href="#cb21-107" tabindex="-1"></a><span class="co">#&gt;                               Pval Pval_group1_vs_group2</span></span>
<span id="cb21-108"><a href="#cb21-108" tabindex="-1"></a><span class="co">#&gt; Pseudomonadota          0.21409442             0.7870908</span></span>
<span id="cb21-109"><a href="#cb21-109" tabindex="-1"></a><span class="co">#&gt; Actinomycetota          0.07304152             0.5632674</span></span>
<span id="cb21-110"><a href="#cb21-110" tabindex="-1"></a><span class="co">#&gt; Thermodesulfobacteriota 0.80778822             0.8845479</span></span>
<span id="cb21-111"><a href="#cb21-111" tabindex="-1"></a><span class="co">#&gt;                         log2FoldChange_group1_vs_group2 Pval_group1_vs_group3</span></span>
<span id="cb21-112"><a href="#cb21-112" tabindex="-1"></a><span class="co">#&gt; Pseudomonadota                               -0.2956677             0.5019108</span></span>
<span id="cb21-113"><a href="#cb21-113" tabindex="-1"></a><span class="co">#&gt; Actinomycetota                               -0.5316898             0.3622041</span></span>
<span id="cb21-114"><a href="#cb21-114" tabindex="-1"></a><span class="co">#&gt; Thermodesulfobacteriota                       0.2995603             0.9863854</span></span>
<span id="cb21-115"><a href="#cb21-115" tabindex="-1"></a><span class="co">#&gt;                         log2FoldChange_group1_vs_group3 Pval_group2_vs_group3</span></span>
<span id="cb21-116"><a href="#cb21-116" tabindex="-1"></a><span class="co">#&gt; Pseudomonadota                               0.71426786            0.19346553</span></span>
<span id="cb21-117"><a href="#cb21-117" tabindex="-1"></a><span class="co">#&gt; Actinomycetota                               1.32040063            0.06022461</span></span>
<span id="cb21-118"><a href="#cb21-118" tabindex="-1"></a><span class="co">#&gt; Thermodesulfobacteriota                     -0.08746284            0.80490509</span></span>
<span id="cb21-119"><a href="#cb21-119" tabindex="-1"></a><span class="co">#&gt;                         log2FoldChange_group2_vs_group3 group1_mean group2_mean</span></span>
<span id="cb21-120"><a href="#cb21-120" tabindex="-1"></a><span class="co">#&gt; Pseudomonadota                                1.0099356      252.25     309.625</span></span>
<span id="cb21-121"><a href="#cb21-121" tabindex="-1"></a><span class="co">#&gt; Actinomycetota                                1.8520905      177.00     255.875</span></span>
<span id="cb21-122"><a href="#cb21-122" tabindex="-1"></a><span class="co">#&gt; Thermodesulfobacteriota                      -0.3870231        2.00       1.625</span></span>
<span id="cb21-123"><a href="#cb21-123" tabindex="-1"></a><span class="co">#&gt;                         group3_mean       test</span></span>
<span id="cb21-124"><a href="#cb21-124" tabindex="-1"></a><span class="co">#&gt; Pseudomonadota              153.750 tukey-test</span></span>
<span id="cb21-125"><a href="#cb21-125" tabindex="-1"></a><span class="co">#&gt; Actinomycetota               70.875 tukey-test</span></span>
<span id="cb21-126"><a href="#cb21-126" tabindex="-1"></a><span class="co">#&gt; Thermodesulfobacteriota       2.125 tukey-test</span></span>
<span id="cb21-127"><a href="#cb21-127" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb21-128"><a href="#cb21-128" tabindex="-1"></a><span class="co">#&gt; $species_abundance</span></span>
<span id="cb21-129"><a href="#cb21-129" tabindex="-1"></a><span class="co">#&gt;                                         Pval Pval_group1_vs_group2</span></span>
<span id="cb21-130"><a href="#cb21-130" tabindex="-1"></a><span class="co">#&gt; Parasutterella excrementihominis 0.199182490             0.8062039</span></span>
<span id="cb21-131"><a href="#cb21-131" tabindex="-1"></a><span class="co">#&gt; Muricaecibacterium torontonense  0.068126536             0.5464864</span></span>
<span id="cb21-132"><a href="#cb21-132" tabindex="-1"></a><span class="co">#&gt; Muribaculaceae bacterium         0.001032886             0.4923547</span></span>
<span id="cb21-133"><a href="#cb21-133" tabindex="-1"></a><span class="co">#&gt;                                  log2FoldChange_group1_vs_group2</span></span>
<span id="cb21-134"><a href="#cb21-134" tabindex="-1"></a><span class="co">#&gt; Parasutterella excrementihominis                      -0.3011695</span></span>
<span id="cb21-135"><a href="#cb21-135" tabindex="-1"></a><span class="co">#&gt; Muricaecibacterium torontonense                       -0.5635121</span></span>
<span id="cb21-136"><a href="#cb21-136" tabindex="-1"></a><span class="co">#&gt; Muribaculaceae bacterium                               0.6818240</span></span>
<span id="cb21-137"><a href="#cb21-137" tabindex="-1"></a><span class="co">#&gt;                                  Pval_group1_vs_group3</span></span>
<span id="cb21-138"><a href="#cb21-138" tabindex="-1"></a><span class="co">#&gt; Parasutterella excrementihominis            0.46121140</span></span>
<span id="cb21-139"><a href="#cb21-139" tabindex="-1"></a><span class="co">#&gt; Muricaecibacterium torontonense             0.35734892</span></span>
<span id="cb21-140"><a href="#cb21-140" tabindex="-1"></a><span class="co">#&gt; Muribaculaceae bacterium                    0.01419407</span></span>
<span id="cb21-141"><a href="#cb21-141" tabindex="-1"></a><span class="co">#&gt;                                  log2FoldChange_group1_vs_group3</span></span>
<span id="cb21-142"><a href="#cb21-142" tabindex="-1"></a><span class="co">#&gt; Parasutterella excrementihominis                       0.8531586</span></span>
<span id="cb21-143"><a href="#cb21-143" tabindex="-1"></a><span class="co">#&gt; Muricaecibacterium torontonense                        1.4364879</span></span>
<span id="cb21-144"><a href="#cb21-144" tabindex="-1"></a><span class="co">#&gt; Muribaculaceae bacterium                              -1.0093379</span></span>
<span id="cb21-145"><a href="#cb21-145" tabindex="-1"></a><span class="co">#&gt;                                  Pval_group2_vs_group3</span></span>
<span id="cb21-146"><a href="#cb21-146" tabindex="-1"></a><span class="co">#&gt; Parasutterella excrementihominis           0.182048409</span></span>
<span id="cb21-147"><a href="#cb21-147" tabindex="-1"></a><span class="co">#&gt; Muricaecibacterium torontonense            0.055825056</span></span>
<span id="cb21-148"><a href="#cb21-148" tabindex="-1"></a><span class="co">#&gt; Muribaculaceae bacterium                   0.000975412</span></span>
<span id="cb21-149"><a href="#cb21-149" tabindex="-1"></a><span class="co">#&gt;                                  log2FoldChange_group2_vs_group3 group1_mean</span></span>
<span id="cb21-150"><a href="#cb21-150" tabindex="-1"></a><span class="co">#&gt; Parasutterella excrementihominis                        1.154328      217.00</span></span>
<span id="cb21-151"><a href="#cb21-151" tabindex="-1"></a><span class="co">#&gt; Muricaecibacterium torontonense                         2.000000      163.75</span></span>
<span id="cb21-152"><a href="#cb21-152" tabindex="-1"></a><span class="co">#&gt; Muribaculaceae bacterium                               -1.691162       38.50</span></span>
<span id="cb21-153"><a href="#cb21-153" tabindex="-1"></a><span class="co">#&gt;                                  group2_mean group3_mean       test</span></span>
<span id="cb21-154"><a href="#cb21-154" tabindex="-1"></a><span class="co">#&gt; Parasutterella excrementihominis     267.375     120.125 tukey-test</span></span>
<span id="cb21-155"><a href="#cb21-155" tabindex="-1"></a><span class="co">#&gt; Muricaecibacterium torontonense      242.000      60.500 tukey-test</span></span>
<span id="cb21-156"><a href="#cb21-156" tabindex="-1"></a><span class="co">#&gt; Muribaculaceae bacterium              24.000      77.500 tukey-test</span></span></code></pre></div>
<p>From here, we could do some further analysis/filtering of the
results. One way we could export these results is as follows. We
write.csv for each of the elements in collated_tables and give it the
same file name as the original abundance table with results_ at the
beginning</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(collated_tables)) {</span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a>  <span class="fu">write.csv</span>(collated_tables[[i]], <span class="fu">paste0</span>(<span class="st">&quot;results_&quot;</span>, a[i]))</span>
<span id="cb22-3"><a href="#cb22-3" tabindex="-1"></a>}</span></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
