---
title: "basic_workflow"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{basic_workflow}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

This basic workflow vignette is intended to demonstate some basic usage of the statomatic package to analyze count data 
from an experiment.

There will be three different analyses included here, meant to demonstrate how statomatic handles different experiment designs. The three 
designs demonstrated here will be a multi group (> 2) two factor design, a two group one factor design, and a multi group one factor design. 

First, load the statomatic package.

```{r setup}
library(statomatic)
```

## Part1: Multi group 2 factor design

The data analyzed here is from an RNA Sequencing experiement which aimed to identify differentially expressed genes across several experiemental 
conditions. The details of this experiment won't be discussed here. 

Load the data to be analyzed. Here we analyze the normalized counts, as created by DESeq2.

```{r data_read-in}
norm_counts <- read.csv("normalized_counts.csv")
head(norm_counts)
```

Notice that the row names of norm_counts are numbers, and that the gene ids are actually inside the data.frame. 
Statomatic will not be able to handle this textual data, so we need to make sure to set the gene ids as the row names 
and remove the gene ids column from inside the data.frame. 

Statomatic has a built in function for this, cf(). This function will set a specified column of a data.frame to be the new row names,
and then remove the column from inside the data.frame. The column which should be the row names can be specified by column_index = an integer, but this 
value defaults to 1 (i.e., the first column). 

```{r column_fix}
norm_counts <- cf(norm_counts)
head(norm_counts)
```

Now the data is ready to be analyzed. 

We also must provide statomatic with metadata about our samples. In the case of this experiment, our samples come from different experimental 
groups. The way to do this is by creating another data.frame which contains our sample names and their experimental conditions. 
We'll use the column names of norm_counts, since these are the samples, and we'll specify conditions of the experiment in additional columns.
The experimental conditions here are sex and genotype. 

```{r make sample info}
sample_info <- data.frame(samples = colnames(norm_counts), sex = c(rep("Male", 10), rep("Female", 10)), 
                          genotype = c(rep("WT", 5), rep("KO", 5), rep("WT", 5), rep("KO", 5)), 
                          group = c(rep("M_WT", 5), rep("M_KO", 5), rep("F_WT", 5), rep("F_KO", 5)))
sample_info
```

Notice that the final column, group, is a linear combination of the first two columns. This is necessary to statomatic's process. The final column of
the sample_info should always describe the distribution of the rest of the experimental conditions (if there are any). 

Now we build the statomatic data set. We'll include the columns, the sample metdata (sample_info), and specify the experimental design.
The experimental design here is sex + genotype. 

```{r build sds part1}
sds <- make_sds(x = norm_counts, colData = sample_info, design = ~ sex + genotype)
sds
```

By viewing sds we can see information about our data such as the number of rows, column names, etc. 

To analyze the data contained in sds, we use the sds_analyze() function. This functions automatically takes information from sds 
in order to perform the analysis. This function will use the experimental design we specified when we created sds

```{r run sds_analyze}
sds <- sds_analyze(sds)
sds
```

Notice that sds appears unchanged. The results of the analysis are stored within sds, and we can access them as follows.

To access all of the p-values from the multi-comparison tests and the fold changes, we can use the get_results() function.

```{r all results}
res <- get_results(sds)
head(res)
```

Notice that in addition to the p-values and fold changes, the mean of each group as well as the test used is at the end of the data.frame. 

If you want to see the anova results, they can be accessed with get_anova().

```{r anova results}
anova_res <- get_anova(sds)
head(anova_res)
```

This data.frame also includes the Tukey multi-comparison results, as well as fold changes.

To access the welch and kruskal-wallis results, use get_welch_test and get_kw. 

You can also access the data.frames ne, nu, and nn by using get_ne, get_nu, and get_nn. 

To see what other results are stored in sds, you can do this:

```{r names}
names(sds@results)
```

You can access any of these results by doing something like sds@results$anova_results.

## Part2: Two group, one factor design

First load the data. For this example we'll look at some gene expression data from an RNA-Seq experiment. These counts were normalized by DESeq2. 
There are two groups in this data set. Group1 is samples 1-4, and group2 is samples 5-8. For the purposes of this example, we'll say that group1 is
"treated" and group2 is "untreated". 

```{r read in}
x <- read.csv("two_group_normalized_counts.csv")
head(x)
```

We need to format the data so that the gene ids in the first column are rownames, and not inside the data.frame. We can use the cf() function for this.

```{r cf}
x <- cf(x)
head(x)
```

Now, we have to make a sample_info object to map our samples to their experimental factors.

```{r sample_info}
sample_info <- data.frame(sample = colnames(x), group = c(rep("treated", 4), rep("untreated", 4)))
sample_info
```

Now we build the statomatic data set

```{r build sds part2}
sds <- make_sds(x, colData = sample_info, design = ~ group)
sds
```

And we run sds_analyze. 

```{r analyze}
sds <- sds_analyze(sds)
```

And we can extract the results by doing: 

```{r get results}
res <- get_results(sds)
head(res)
```

You can access most results stored in sds with a function that starts with get_ and ends with the name of the result.
To see a full list of results stored in sds, you can do this:

```{r @results}
names(sds@results)
```

And any of these can be accessed by sds@results$result_name

Notice that we used the same function (sds_analyze) to analyze both multi-group and two-group
data. The sds_analyze function uses the design information supplied during the creation of sds to decide whether to apply two-group or multi-group
tests. 

We're going to use the same sds_analyze function again for the next part of this vignette. 

## Part3: Multi-group one factor design

The data we use here is some species level data from a metagenomic analysis. There are 24 samples, and three experimental factors with 8 samples 
per group. For our purposes here, we'll call the factors group1, 2, and 3. The pace of this analysis will be a bit faster than the previous one,
as most steps are very similar. 

```{r}
#read in data
x <- read.csv("species_abundance.csv")
#fix rownames
x <- cf(x)
head(x)
```

```{r}
#make sample_info
sample_info <- data.frame(sample = colnames(x), group = c(rep("group1", 8), rep("group2", 8), rep("group3", 8)))
sample_info
```

```{r}
#make sds
sds <- make_sds(x, colData = sample_info, design = ~ group)
sds <- sds_analyze(sds)
```

Notice here that statomatic tells us that it's using a one way anova to analyze the normal-equal variance portion of the data.
This is because of our specified design. 

```{r}
res <- get_results(sds)
head(res)
```
