<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>basic_workflow</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">basic_workflow</h1>



<p>This basic workflow vignette is intended to demonstate some basic
usage of the statomatic package to analyze count data from an
experiment.</p>
<p>There will be three different analyses included here, meant to
demonstrate how statomatic handles different experiment designs. The
three designs demonstrated here will be a multi group (&gt; 2) two
factor design, a two group one factor design, and a multi group one
factor design.</p>
<p>First, load the statomatic package.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(statomatic)</span></code></pre></div>
<div id="part1-multi-group-2-factor-design" class="section level2">
<h2>Part1: Multi group 2 factor design</h2>
<p>The data analyzed here is from an RNA Sequencing experiement which
aimed to identify differentially expressed genes across several
experiemental conditions. The details of this experiment won’t be
discussed here.</p>
<p>Load the data to be analyzed. Here we analyze the normalized counts,
as created by DESeq2.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>norm_counts <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;normalized_counts.csv&quot;</span>)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">head</span>(norm_counts)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="co">#&gt;                    X Sample_600 Sample_601 Sample_602 Sample_616 Sample_617</span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="co">#&gt; 1 ENSMUSG00000051285  169.21935  309.83231 417.709088  214.03944  202.50044</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="co">#&gt; 2 ENSMUSG00000057363   47.53353   66.64997  54.805802   41.23696   49.30862</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="co">#&gt; 3 ENSMUSG00000100555   23.76676   55.84187   0.000000   48.10978   18.19153</span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="co">#&gt; 4 ENSMUSG00000078184   50.38554   59.44457   5.924952   12.76382   24.41495</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="co">#&gt; 5 ENSMUSG00000061024  273.79311  293.62015 232.554350  144.32935  360.00079</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a><span class="co">#&gt; 6 ENSMUSG00000025911  154.95929  525.99438 309.578721  208.14845  256.59631</span></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a><span class="co">#&gt;   Sample_603 Sample_604 Sample_605 Sample_606 Sample_621 Sample_607 Sample_608</span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a><span class="co">#&gt; 1  110.24045  171.38160   61.79141 190.642528  427.33305  228.73359  410.89116</span></span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a><span class="co">#&gt; 2   57.16171    0.00000    0.00000 103.558904   58.42021   63.76202   88.17407</span></span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a><span class="co">#&gt; 3   28.58086  151.21906   68.65713   5.884029   30.29196   21.25401   15.87133</span></span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a><span class="co">#&gt; 4   32.66384   10.08127    0.00000  44.718618   37.86495   19.22981   18.51655</span></span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a><span class="co">#&gt; 5  145.62627   55.44699  109.85140 222.416283  315.90190  253.02388  319.19013</span></span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a><span class="co">#&gt; 6  155.15322  171.38160    0.00000 264.781289  529.02750  301.60446  336.82494</span></span>
<span id="cb2-17"><a href="#cb2-17" tabindex="-1"></a><span class="co">#&gt;   Sample_609 Sample_610 Sample_623 Sample_611 Sample_612 Sample_613 Sample_614</span></span>
<span id="cb2-18"><a href="#cb2-18" tabindex="-1"></a><span class="co">#&gt; 1  313.05044  412.25499  133.76758  93.303195 248.040131   46.21469  417.35080</span></span>
<span id="cb2-19"><a href="#cb2-19" tabindex="-1"></a><span class="co">#&gt; 2   60.75650   14.45406   99.46820 103.366522  18.445254    0.00000   65.25144</span></span>
<span id="cb2-20"><a href="#cb2-20" tabindex="-1"></a><span class="co">#&gt; 3   45.30993   40.01052   30.86944   5.093783   1.456204  390.25738    0.00000</span></span>
<span id="cb2-21"><a href="#cb2-21" tabindex="-1"></a><span class="co">#&gt; 4   39.13131   36.86833   56.59398  57.025521  49.996347    0.00000   82.04884</span></span>
<span id="cb2-22"><a href="#cb2-22" tabindex="-1"></a><span class="co">#&gt; 5  209.04355  211.36447  221.23100 240.525945 248.040131  302.96296  440.60874</span></span>
<span id="cb2-23"><a href="#cb2-23" tabindex="-1"></a><span class="co">#&gt; 6  375.86649  528.09696  265.82019 148.589375 171.832106   10.26993  323.02694</span></span>
<span id="cb2-24"><a href="#cb2-24" tabindex="-1"></a><span class="co">#&gt;   Sample_615</span></span>
<span id="cb2-25"><a href="#cb2-25" tabindex="-1"></a><span class="co">#&gt; 1  513.48498</span></span>
<span id="cb2-26"><a href="#cb2-26" tabindex="-1"></a><span class="co">#&gt; 2   39.36603</span></span>
<span id="cb2-27"><a href="#cb2-27" tabindex="-1"></a><span class="co">#&gt; 3    0.00000</span></span>
<span id="cb2-28"><a href="#cb2-28" tabindex="-1"></a><span class="co">#&gt; 4   14.61838</span></span>
<span id="cb2-29"><a href="#cb2-29" tabindex="-1"></a><span class="co">#&gt; 5  213.52043</span></span>
<span id="cb2-30"><a href="#cb2-30" tabindex="-1"></a><span class="co">#&gt; 6  646.77697</span></span></code></pre></div>
<p>Notice that the row names of norm_counts are numbers, and that the
gene ids are actually inside the data.frame. Statomatic will not be able
to handle this textual data, so we need to make sure to set the gene ids
as the row names and remove the gene ids column from inside the
data.frame.</p>
<p>Statomatic has a built in function for this, cf(). This function will
set a specified column of a data.frame to be the new row names, and then
remove the column from inside the data.frame. The column which should be
the row names can be specified by column_index = an integer, but this
value defaults to 1 (i.e., the first column).</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>norm_counts <span class="ot">&lt;-</span> <span class="fu">cf</span>(norm_counts)</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="fu">head</span>(norm_counts)</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="co">#&gt;                    Sample_600 Sample_601 Sample_602 Sample_616 Sample_617</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051285  169.21935  309.83231 417.709088  214.03944  202.50044</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057363   47.53353   66.64997  54.805802   41.23696   49.30862</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000100555   23.76676   55.84187   0.000000   48.10978   18.19153</span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000078184   50.38554   59.44457   5.924952   12.76382   24.41495</span></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000061024  273.79311  293.62015 232.554350  144.32935  360.00079</span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025911  154.95929  525.99438 309.578721  208.14845  256.59631</span></span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a><span class="co">#&gt;                    Sample_603 Sample_604 Sample_605 Sample_606 Sample_621</span></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051285  110.24045  171.38160   61.79141 190.642528  427.33305</span></span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057363   57.16171    0.00000    0.00000 103.558904   58.42021</span></span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000100555   28.58086  151.21906   68.65713   5.884029   30.29196</span></span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000078184   32.66384   10.08127    0.00000  44.718618   37.86495</span></span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000061024  145.62627   55.44699  109.85140 222.416283  315.90190</span></span>
<span id="cb3-16"><a href="#cb3-16" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025911  155.15322  171.38160    0.00000 264.781289  529.02750</span></span>
<span id="cb3-17"><a href="#cb3-17" tabindex="-1"></a><span class="co">#&gt;                    Sample_607 Sample_608 Sample_609 Sample_610 Sample_623</span></span>
<span id="cb3-18"><a href="#cb3-18" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051285  228.73359  410.89116  313.05044  412.25499  133.76758</span></span>
<span id="cb3-19"><a href="#cb3-19" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057363   63.76202   88.17407   60.75650   14.45406   99.46820</span></span>
<span id="cb3-20"><a href="#cb3-20" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000100555   21.25401   15.87133   45.30993   40.01052   30.86944</span></span>
<span id="cb3-21"><a href="#cb3-21" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000078184   19.22981   18.51655   39.13131   36.86833   56.59398</span></span>
<span id="cb3-22"><a href="#cb3-22" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000061024  253.02388  319.19013  209.04355  211.36447  221.23100</span></span>
<span id="cb3-23"><a href="#cb3-23" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025911  301.60446  336.82494  375.86649  528.09696  265.82019</span></span>
<span id="cb3-24"><a href="#cb3-24" tabindex="-1"></a><span class="co">#&gt;                    Sample_611 Sample_612 Sample_613 Sample_614 Sample_615</span></span>
<span id="cb3-25"><a href="#cb3-25" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051285  93.303195 248.040131   46.21469  417.35080  513.48498</span></span>
<span id="cb3-26"><a href="#cb3-26" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057363 103.366522  18.445254    0.00000   65.25144   39.36603</span></span>
<span id="cb3-27"><a href="#cb3-27" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000100555   5.093783   1.456204  390.25738    0.00000    0.00000</span></span>
<span id="cb3-28"><a href="#cb3-28" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000078184  57.025521  49.996347    0.00000   82.04884   14.61838</span></span>
<span id="cb3-29"><a href="#cb3-29" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000061024 240.525945 248.040131  302.96296  440.60874  213.52043</span></span>
<span id="cb3-30"><a href="#cb3-30" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025911 148.589375 171.832106   10.26993  323.02694  646.77697</span></span></code></pre></div>
<p>Now the data is ready to be analyzed.</p>
<p>We also must provide statomatic with metadata about our samples. In
the case of this experiment, our samples come from different
experimental groups. The way to do this is by creating another
data.frame which contains our sample names and their experimental
conditions. We’ll use the column names of norm_counts, since these are
the samples, and we’ll specify conditions of the experiment in
additional columns. The experimental conditions here are sex and
genotype.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>sample_info <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">samples =</span> <span class="fu">colnames</span>(norm_counts), <span class="at">sex =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;Male&quot;</span>, <span class="dv">10</span>), <span class="fu">rep</span>(<span class="st">&quot;Female&quot;</span>, <span class="dv">10</span>)), </span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>                          <span class="at">genotype =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;WT&quot;</span>, <span class="dv">5</span>), <span class="fu">rep</span>(<span class="st">&quot;KO&quot;</span>, <span class="dv">5</span>), <span class="fu">rep</span>(<span class="st">&quot;WT&quot;</span>, <span class="dv">5</span>), <span class="fu">rep</span>(<span class="st">&quot;KO&quot;</span>, <span class="dv">5</span>)), </span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>                          <span class="at">group =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;M_WT&quot;</span>, <span class="dv">5</span>), <span class="fu">rep</span>(<span class="st">&quot;M_KO&quot;</span>, <span class="dv">5</span>), <span class="fu">rep</span>(<span class="st">&quot;F_WT&quot;</span>, <span class="dv">5</span>), <span class="fu">rep</span>(<span class="st">&quot;F_KO&quot;</span>, <span class="dv">5</span>)))</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>sample_info</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a><span class="co">#&gt;       samples    sex genotype group</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a><span class="co">#&gt; 1  Sample_600   Male       WT  M_WT</span></span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a><span class="co">#&gt; 2  Sample_601   Male       WT  M_WT</span></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a><span class="co">#&gt; 3  Sample_602   Male       WT  M_WT</span></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a><span class="co">#&gt; 4  Sample_616   Male       WT  M_WT</span></span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a><span class="co">#&gt; 5  Sample_617   Male       WT  M_WT</span></span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a><span class="co">#&gt; 6  Sample_603   Male       KO  M_KO</span></span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a><span class="co">#&gt; 7  Sample_604   Male       KO  M_KO</span></span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a><span class="co">#&gt; 8  Sample_605   Male       KO  M_KO</span></span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a><span class="co">#&gt; 9  Sample_606   Male       KO  M_KO</span></span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a><span class="co">#&gt; 10 Sample_621   Male       KO  M_KO</span></span>
<span id="cb4-16"><a href="#cb4-16" tabindex="-1"></a><span class="co">#&gt; 11 Sample_607 Female       WT  F_WT</span></span>
<span id="cb4-17"><a href="#cb4-17" tabindex="-1"></a><span class="co">#&gt; 12 Sample_608 Female       WT  F_WT</span></span>
<span id="cb4-18"><a href="#cb4-18" tabindex="-1"></a><span class="co">#&gt; 13 Sample_609 Female       WT  F_WT</span></span>
<span id="cb4-19"><a href="#cb4-19" tabindex="-1"></a><span class="co">#&gt; 14 Sample_610 Female       WT  F_WT</span></span>
<span id="cb4-20"><a href="#cb4-20" tabindex="-1"></a><span class="co">#&gt; 15 Sample_623 Female       WT  F_WT</span></span>
<span id="cb4-21"><a href="#cb4-21" tabindex="-1"></a><span class="co">#&gt; 16 Sample_611 Female       KO  F_KO</span></span>
<span id="cb4-22"><a href="#cb4-22" tabindex="-1"></a><span class="co">#&gt; 17 Sample_612 Female       KO  F_KO</span></span>
<span id="cb4-23"><a href="#cb4-23" tabindex="-1"></a><span class="co">#&gt; 18 Sample_613 Female       KO  F_KO</span></span>
<span id="cb4-24"><a href="#cb4-24" tabindex="-1"></a><span class="co">#&gt; 19 Sample_614 Female       KO  F_KO</span></span>
<span id="cb4-25"><a href="#cb4-25" tabindex="-1"></a><span class="co">#&gt; 20 Sample_615 Female       KO  F_KO</span></span></code></pre></div>
<p>Notice that the final column, group, is a linear combination of the
first two columns. This is necessary to statomatic’s process. The final
column of the sample_info should always describe the distribution of the
rest of the experimental conditions (if there are any).</p>
<p>Now we build the statomatic data set. We’ll include the columns, the
sample metdata (sample_info), and specify the experimental design. The
experimental design here is sex + genotype.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>sds <span class="ot">&lt;-</span> <span class="fu">make_sds</span>(<span class="at">x =</span> norm_counts, <span class="at">colData =</span> sample_info, <span class="at">design =</span> <span class="sc">~</span> sex <span class="sc">+</span> genotype)</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>sds</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="co">#&gt; class: StatomaticDataset </span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="co">#&gt; dim: 100 20 </span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a><span class="co">#&gt; metadata(0):</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a><span class="co">#&gt; assays(1): counts</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a><span class="co">#&gt; rownames(100): ENSMUSG00000051285 ENSMUSG00000057363 ...</span></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a><span class="co">#&gt;   ENSMUSG00000026384 ENSMUSG00000036975</span></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a><span class="co">#&gt; rowData names(0):</span></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a><span class="co">#&gt; colnames(20): Sample_600 Sample_601 ... Sample_614 Sample_615</span></span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a><span class="co">#&gt; colData names(4): samples sex genotype group</span></span></code></pre></div>
<p>By viewing sds we can see information about our data such as the
number of rows, column names, etc.</p>
<p>To analyze the data contained in sds, we use the sds_analyze()
function. This functions automatically takes information from sds in
order to perform the analysis. This function will use the experimental
design we specified when we created sds</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>sds <span class="ot">&lt;-</span> <span class="fu">sds_analyze</span>(sds)</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="co">#&gt; sorting data 100% done...</span></span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="co">#&gt; Using two way anova </span></span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a><span class="co">#&gt; anova 100% done</span></span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a><span class="co">#&gt; Tukey test 100% done</span></span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a><span class="co">#&gt; Welch test 100% done</span></span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a><span class="co">#&gt; DunnettT3 test 100% done</span></span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a><span class="co">#&gt; Kruskal-Wallis Test 100% done</span></span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a><span class="co">#&gt; Dunn Test 100% done</span></span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb6-13"><a href="#cb6-13" tabindex="-1"></a>sds</span>
<span id="cb6-14"><a href="#cb6-14" tabindex="-1"></a><span class="co">#&gt; class: StatomaticDataset </span></span>
<span id="cb6-15"><a href="#cb6-15" tabindex="-1"></a><span class="co">#&gt; dim: 100 20 </span></span>
<span id="cb6-16"><a href="#cb6-16" tabindex="-1"></a><span class="co">#&gt; metadata(0):</span></span>
<span id="cb6-17"><a href="#cb6-17" tabindex="-1"></a><span class="co">#&gt; assays(1): counts</span></span>
<span id="cb6-18"><a href="#cb6-18" tabindex="-1"></a><span class="co">#&gt; rownames(100): ENSMUSG00000051285 ENSMUSG00000057363 ...</span></span>
<span id="cb6-19"><a href="#cb6-19" tabindex="-1"></a><span class="co">#&gt;   ENSMUSG00000026384 ENSMUSG00000036975</span></span>
<span id="cb6-20"><a href="#cb6-20" tabindex="-1"></a><span class="co">#&gt; rowData names(0):</span></span>
<span id="cb6-21"><a href="#cb6-21" tabindex="-1"></a><span class="co">#&gt; colnames(20): Sample_600 Sample_601 ... Sample_614 Sample_615</span></span>
<span id="cb6-22"><a href="#cb6-22" tabindex="-1"></a><span class="co">#&gt; colData names(4): samples sex genotype group</span></span></code></pre></div>
<p>Notice that sds appears unchanged. The results of the analysis are
stored within sds, and we can access them as follows.</p>
<p>To access all of the p-values from the multi-comparison tests and the
fold changes, we can use the get_results() function.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">get_results</span>(sds)</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="fu">head</span>(res)</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="co">#&gt;                    Pval_F_KO_vs_F_WT log2FoldChange_F_KO_vs_F_WT</span></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051285         0.9790520                  -0.1849281</span></span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057363         0.7955281                  -0.5285301</span></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000078184         0.9699575                   0.2579516</span></span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000061024         0.8094858                   0.2521327</span></span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025911         0.8062227                  -0.4755035</span></span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057335         0.8020005                  -0.5990134</span></span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a><span class="co">#&gt;                    Pval_F_KO_vs_M_KO log2FoldChange_F_KO_vs_M_KO</span></span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051285         0.8652082                   0.4555891</span></span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057363         0.9998882                   0.0472020</span></span>
<span id="cb7-13"><a href="#cb7-13" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000078184         0.7280883                   0.7006521</span></span>
<span id="cb7-14"><a href="#cb7-14" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000061024         0.1413002                   0.7674774</span></span>
<span id="cb7-15"><a href="#cb7-15" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025911         0.9883949                   0.2151199</span></span>
<span id="cb7-16"><a href="#cb7-16" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057335         0.9992630                   0.1067049</span></span>
<span id="cb7-17"><a href="#cb7-17" tabindex="-1"></a><span class="co">#&gt;                    Pval_F_KO_vs_M_WT log2FoldChange_F_KO_vs_M_WT</span></span>
<span id="cb7-18"><a href="#cb7-18" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051285         0.9999995                 0.005584147</span></span>
<span id="cb7-19"><a href="#cb7-19" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057363         0.9899203                -0.196868100</span></span>
<span id="cb7-20"><a href="#cb7-20" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000078184         0.9050970                 0.413461170</span></span>
<span id="cb7-21"><a href="#cb7-21" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000061024         0.9469754                 0.148453255</span></span>
<span id="cb7-22"><a href="#cb7-22" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025911         0.9925533                -0.162232725</span></span>
<span id="cb7-23"><a href="#cb7-23" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057335         0.7267167                -0.672042886</span></span>
<span id="cb7-24"><a href="#cb7-24" tabindex="-1"></a><span class="co">#&gt;                    Pval_F_WT_vs_M_KO log2FoldChange_F_WT_vs_M_KO</span></span>
<span id="cb7-25"><a href="#cb7-25" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051285         0.6568296                   0.6405172</span></span>
<span id="cb7-26"><a href="#cb7-26" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057363         0.7597698                   0.5757321</span></span>
<span id="cb7-27"><a href="#cb7-27" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000078184         0.9310571                   0.4427005</span></span>
<span id="cb7-28"><a href="#cb7-28" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000061024         0.5161829                   0.5153447</span></span>
<span id="cb7-29"><a href="#cb7-29" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025911         0.6260176                   0.6906234</span></span>
<span id="cb7-30"><a href="#cb7-30" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057335         0.7340525                   0.7057183</span></span>
<span id="cb7-31"><a href="#cb7-31" tabindex="-1"></a><span class="co">#&gt;                    Pval_F_WT_vs_M_WT log2FoldChange_F_WT_vs_M_WT</span></span>
<span id="cb7-32"><a href="#cb7-32" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051285         0.9773126                  0.19051226</span></span>
<span id="cb7-33"><a href="#cb7-33" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057363         0.9258447                  0.33166197</span></span>
<span id="cb7-34"><a href="#cb7-34" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000078184         0.9954529                  0.15550958</span></span>
<span id="cb7-35"><a href="#cb7-35" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000061024         0.9850262                 -0.10367949</span></span>
<span id="cb7-36"><a href="#cb7-36" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025911         0.9228793                  0.31327074</span></span>
<span id="cb7-37"><a href="#cb7-37" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057335         0.9990121                 -0.07302953</span></span>
<span id="cb7-38"><a href="#cb7-38" tabindex="-1"></a><span class="co">#&gt;                    Pval_M_KO_vs_M_WT log2FoldChange_M_KO_vs_M_WT F_KO_mean</span></span>
<span id="cb7-39"><a href="#cb7-39" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051285         0.8699999                  -0.4500049 263.67876</span></span>
<span id="cb7-40"><a href="#cb7-40" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057363         0.9820468                  -0.2440701  45.28585</span></span>
<span id="cb7-41"><a href="#cb7-41" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000078184         0.9824965                  -0.2871910  40.73782</span></span>
<span id="cb7-42"><a href="#cb7-42" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000061024         0.3321566                  -0.6190242 289.13164</span></span>
<span id="cb7-43"><a href="#cb7-43" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025911         0.9330056                  -0.3773527 260.09906</span></span>
<span id="cb7-44"><a href="#cb7-44" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057335         0.6535879                  -0.7787478 189.02091</span></span>
<span id="cb7-45"><a href="#cb7-45" tabindex="-1"></a><span class="co">#&gt;                    F_WT_mean M_KO_mean M_WT_mean       test</span></span>
<span id="cb7-46"><a href="#cb7-46" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051285 299.73955 192.27781 262.66013 tukey-test</span></span>
<span id="cb7-47"><a href="#cb7-47" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057363  65.32297  43.82817  51.90698 tukey-test</span></span>
<span id="cb7-48"><a href="#cb7-48" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000078184  34.06800  25.06574  30.58677 tukey-test</span></span>
<span id="cb7-49"><a href="#cb7-49" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000061024 242.77061 169.84857 260.85955 tukey-test</span></span>
<span id="cb7-50"><a href="#cb7-50" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025911 361.64261 224.06872 291.05543 tukey-test</span></span>
<span id="cb7-51"><a href="#cb7-51" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057335 286.30626 175.54500 301.17223 tukey-test</span></span></code></pre></div>
<p>Notice that in addition to the p-values and fold changes, the mean of
each group as well as the test used is at the end of the data.frame.</p>
<p>If you want to see the anova results, they can be accessed with
get_anova().</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>anova_res <span class="ot">&lt;-</span> <span class="fu">get_anova</span>(sds)</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="fu">head</span>(anova_res)</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a><span class="co">#&gt;                          sex  genotype interaction Pval_F_KO_vs_F_WT</span></span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051285 0.4181785 0.4267951  0.79596064         0.9790520</span></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057363 0.6363190 0.3757298  0.70346317         0.7955281</span></span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000078184 0.3813036 0.9576038  0.57458343         0.9699575</span></span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000061024 0.1879463 0.5524404  0.08030823         0.8094858</span></span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025911 0.5146527 0.3078191  0.83172345         0.8062227</span></span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057335 0.9928111 0.1615905  0.85432131         0.8020005</span></span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a><span class="co">#&gt;                    log2FoldChange_F_KO_vs_F_WT Pval_F_KO_vs_M_KO</span></span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051285                  -0.1849281         0.8652082</span></span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057363                  -0.5285301         0.9998882</span></span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000078184                   0.2579516         0.7280883</span></span>
<span id="cb8-14"><a href="#cb8-14" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000061024                   0.2521327         0.1413002</span></span>
<span id="cb8-15"><a href="#cb8-15" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025911                  -0.4755035         0.9883949</span></span>
<span id="cb8-16"><a href="#cb8-16" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057335                  -0.5990134         0.9992630</span></span>
<span id="cb8-17"><a href="#cb8-17" tabindex="-1"></a><span class="co">#&gt;                    log2FoldChange_F_KO_vs_M_KO Pval_F_KO_vs_M_WT</span></span>
<span id="cb8-18"><a href="#cb8-18" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051285                   0.4555891         0.9999995</span></span>
<span id="cb8-19"><a href="#cb8-19" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057363                   0.0472020         0.9899203</span></span>
<span id="cb8-20"><a href="#cb8-20" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000078184                   0.7006521         0.9050970</span></span>
<span id="cb8-21"><a href="#cb8-21" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000061024                   0.7674774         0.9469754</span></span>
<span id="cb8-22"><a href="#cb8-22" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025911                   0.2151199         0.9925533</span></span>
<span id="cb8-23"><a href="#cb8-23" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057335                   0.1067049         0.7267167</span></span>
<span id="cb8-24"><a href="#cb8-24" tabindex="-1"></a><span class="co">#&gt;                    log2FoldChange_F_KO_vs_M_WT Pval_F_WT_vs_M_KO</span></span>
<span id="cb8-25"><a href="#cb8-25" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051285                 0.005584147         0.6568296</span></span>
<span id="cb8-26"><a href="#cb8-26" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057363                -0.196868100         0.7597698</span></span>
<span id="cb8-27"><a href="#cb8-27" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000078184                 0.413461170         0.9310571</span></span>
<span id="cb8-28"><a href="#cb8-28" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000061024                 0.148453255         0.5161829</span></span>
<span id="cb8-29"><a href="#cb8-29" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025911                -0.162232725         0.6260176</span></span>
<span id="cb8-30"><a href="#cb8-30" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057335                -0.672042886         0.7340525</span></span>
<span id="cb8-31"><a href="#cb8-31" tabindex="-1"></a><span class="co">#&gt;                    log2FoldChange_F_WT_vs_M_KO Pval_F_WT_vs_M_WT</span></span>
<span id="cb8-32"><a href="#cb8-32" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051285                   0.6405172         0.9773126</span></span>
<span id="cb8-33"><a href="#cb8-33" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057363                   0.5757321         0.9258447</span></span>
<span id="cb8-34"><a href="#cb8-34" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000078184                   0.4427005         0.9954529</span></span>
<span id="cb8-35"><a href="#cb8-35" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000061024                   0.5153447         0.9850262</span></span>
<span id="cb8-36"><a href="#cb8-36" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025911                   0.6906234         0.9228793</span></span>
<span id="cb8-37"><a href="#cb8-37" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057335                   0.7057183         0.9990121</span></span>
<span id="cb8-38"><a href="#cb8-38" tabindex="-1"></a><span class="co">#&gt;                    log2FoldChange_F_WT_vs_M_WT Pval_M_KO_vs_M_WT</span></span>
<span id="cb8-39"><a href="#cb8-39" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051285                  0.19051226         0.8699999</span></span>
<span id="cb8-40"><a href="#cb8-40" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057363                  0.33166197         0.9820468</span></span>
<span id="cb8-41"><a href="#cb8-41" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000078184                  0.15550958         0.9824965</span></span>
<span id="cb8-42"><a href="#cb8-42" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000061024                 -0.10367949         0.3321566</span></span>
<span id="cb8-43"><a href="#cb8-43" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025911                  0.31327074         0.9330056</span></span>
<span id="cb8-44"><a href="#cb8-44" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057335                 -0.07302953         0.6535879</span></span>
<span id="cb8-45"><a href="#cb8-45" tabindex="-1"></a><span class="co">#&gt;                    log2FoldChange_M_KO_vs_M_WT F_KO_mean F_WT_mean M_KO_mean</span></span>
<span id="cb8-46"><a href="#cb8-46" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051285                  -0.4500049 263.67876 299.73955 192.27781</span></span>
<span id="cb8-47"><a href="#cb8-47" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057363                  -0.2440701  45.28585  65.32297  43.82817</span></span>
<span id="cb8-48"><a href="#cb8-48" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000078184                  -0.2871910  40.73782  34.06800  25.06574</span></span>
<span id="cb8-49"><a href="#cb8-49" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000061024                  -0.6190242 289.13164 242.77061 169.84857</span></span>
<span id="cb8-50"><a href="#cb8-50" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025911                  -0.3773527 260.09906 361.64261 224.06872</span></span>
<span id="cb8-51"><a href="#cb8-51" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057335                  -0.7787478 189.02091 286.30626 175.54500</span></span>
<span id="cb8-52"><a href="#cb8-52" tabindex="-1"></a><span class="co">#&gt;                    M_WT_mean       test</span></span>
<span id="cb8-53"><a href="#cb8-53" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051285 262.66013 tukey-test</span></span>
<span id="cb8-54"><a href="#cb8-54" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057363  51.90698 tukey-test</span></span>
<span id="cb8-55"><a href="#cb8-55" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000078184  30.58677 tukey-test</span></span>
<span id="cb8-56"><a href="#cb8-56" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000061024 260.85955 tukey-test</span></span>
<span id="cb8-57"><a href="#cb8-57" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025911 291.05543 tukey-test</span></span>
<span id="cb8-58"><a href="#cb8-58" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000057335 301.17223 tukey-test</span></span></code></pre></div>
<p>This data.frame also includes the Tukey multi-comparison results, as
well as fold changes.</p>
<p>To access the welch and kruskal-wallis results, use get_welch_test
and get_kw.</p>
<p>You can also access the data.frames ne, nu, and nn by using get_ne,
get_nu, and get_nn. </p>
<p>To see what other results are stored in sds, you can do this:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="fu">names</span>(sds<span class="sc">@</span>results)</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a><span class="co">#&gt; [1] &quot;anova_results&quot;      &quot;welch_test_results&quot; &quot;kw_results&quot;        </span></span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a><span class="co">#&gt; [4] &quot;all_results&quot;        &quot;ne&quot;                 &quot;nu&quot;                </span></span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a><span class="co">#&gt; [7] &quot;nn&quot;                 &quot;foldchange&quot;</span></span></code></pre></div>
<p>You can access any of these results by doing something like <a href="mailto:sds@results" class="email">sds@results</a>$anova_results.</p>
</div>
<div id="part2-two-group-one-factor-design" class="section level2">
<h2>Part2: Two group, one factor design</h2>
<p>First load the data. For this example we’ll look at some gene
expression data from an RNA-Seq experiment. These counts were normalized
by DESeq2. There are two groups in this data set. Group1 is samples 1-4,
and group2 is samples 5-8. For the purposes of this example, we’ll say
that group1 is “treated” and group2 is “untreated”.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;two_group_normalized_counts.csv&quot;</span>)</span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a><span class="fu">head</span>(x)</span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a><span class="co">#&gt;                    X   sample_1   sample_2   sample_3   sample_4   sample_5</span></span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a><span class="co">#&gt; 1 ENSMUSG00000000001 1707.89280 1843.60723 1687.02686 1501.02388 1806.65191</span></span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a><span class="co">#&gt; 2 ENSMUSG00000000028   27.84608   30.17596   29.61100   69.66777   48.57703</span></span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a><span class="co">#&gt; 3 ENSMUSG00000000056  468.74232  517.30211  556.20667  404.54810  455.79719</span></span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a><span class="co">#&gt; 4 ENSMUSG00000000058   60.33317   81.90617   75.22795   51.45915  116.79157</span></span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a><span class="co">#&gt; 5 ENSMUSG00000000078 2706.87086 3110.99739 2504.93077 2310.91176 2468.12630</span></span>
<span id="cb10-9"><a href="#cb10-9" tabindex="-1"></a><span class="co">#&gt; 6 ENSMUSG00000000085  158.95470  145.13198  226.48416  192.37806  248.05289</span></span>
<span id="cb10-10"><a href="#cb10-10" tabindex="-1"></a><span class="co">#&gt;     sample_6   sample_7   sample_8</span></span>
<span id="cb10-11"><a href="#cb10-11" tabindex="-1"></a><span class="co">#&gt; 1 1916.63101 1984.70503 2100.84931</span></span>
<span id="cb10-12"><a href="#cb10-12" tabindex="-1"></a><span class="co">#&gt; 2   51.32071   42.77382   52.52123</span></span>
<span id="cb10-13"><a href="#cb10-13" tabindex="-1"></a><span class="co">#&gt; 3  468.79492  497.03174  504.00183</span></span>
<span id="cb10-14"><a href="#cb10-14" tabindex="-1"></a><span class="co">#&gt; 4   88.82430   71.86001   68.68161</span></span>
<span id="cb10-15"><a href="#cb10-15" tabindex="-1"></a><span class="co">#&gt; 5 2762.43574 2557.01869 2697.77332</span></span>
<span id="cb10-16"><a href="#cb10-16" tabindex="-1"></a><span class="co">#&gt; 6  224.03462  203.60336  198.97467</span></span></code></pre></div>
<p>We need to format the data so that the gene ids in the first column
are rownames, and not inside the data.frame. We can use the cf()
function for this.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">cf</span>(x)</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a><span class="fu">head</span>(x)</span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a><span class="co">#&gt;                      sample_1   sample_2   sample_3   sample_4   sample_5</span></span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000000001 1707.89280 1843.60723 1687.02686 1501.02388 1806.65191</span></span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000000028   27.84608   30.17596   29.61100   69.66777   48.57703</span></span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000000056  468.74232  517.30211  556.20667  404.54810  455.79719</span></span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000000058   60.33317   81.90617   75.22795   51.45915  116.79157</span></span>
<span id="cb11-8"><a href="#cb11-8" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000000078 2706.87086 3110.99739 2504.93077 2310.91176 2468.12630</span></span>
<span id="cb11-9"><a href="#cb11-9" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000000085  158.95470  145.13198  226.48416  192.37806  248.05289</span></span>
<span id="cb11-10"><a href="#cb11-10" tabindex="-1"></a><span class="co">#&gt;                      sample_6   sample_7   sample_8</span></span>
<span id="cb11-11"><a href="#cb11-11" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000000001 1916.63101 1984.70503 2100.84931</span></span>
<span id="cb11-12"><a href="#cb11-12" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000000028   51.32071   42.77382   52.52123</span></span>
<span id="cb11-13"><a href="#cb11-13" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000000056  468.79492  497.03174  504.00183</span></span>
<span id="cb11-14"><a href="#cb11-14" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000000058   88.82430   71.86001   68.68161</span></span>
<span id="cb11-15"><a href="#cb11-15" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000000078 2762.43574 2557.01869 2697.77332</span></span>
<span id="cb11-16"><a href="#cb11-16" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000000085  224.03462  203.60336  198.97467</span></span></code></pre></div>
<p>Now, we have to make a sample_info object to map our samples to their
experimental factors.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a>sample_info <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">sample =</span> <span class="fu">colnames</span>(x), <span class="at">group =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;treated&quot;</span>, <span class="dv">4</span>), <span class="fu">rep</span>(<span class="st">&quot;untreated&quot;</span>, <span class="dv">4</span>)))</span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a>sample_info</span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a><span class="co">#&gt;     sample     group</span></span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a><span class="co">#&gt; 1 sample_1   treated</span></span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a><span class="co">#&gt; 2 sample_2   treated</span></span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a><span class="co">#&gt; 3 sample_3   treated</span></span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a><span class="co">#&gt; 4 sample_4   treated</span></span>
<span id="cb12-8"><a href="#cb12-8" tabindex="-1"></a><span class="co">#&gt; 5 sample_5 untreated</span></span>
<span id="cb12-9"><a href="#cb12-9" tabindex="-1"></a><span class="co">#&gt; 6 sample_6 untreated</span></span>
<span id="cb12-10"><a href="#cb12-10" tabindex="-1"></a><span class="co">#&gt; 7 sample_7 untreated</span></span>
<span id="cb12-11"><a href="#cb12-11" tabindex="-1"></a><span class="co">#&gt; 8 sample_8 untreated</span></span></code></pre></div>
<p>Now we build the statomatic data set</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a>sds <span class="ot">&lt;-</span> <span class="fu">make_sds</span>(x, <span class="at">colData =</span> sample_info, <span class="at">design =</span> <span class="sc">~</span> group)</span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>sds</span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a><span class="co">#&gt; class: StatomaticDataset </span></span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a><span class="co">#&gt; dim: 100 8 </span></span>
<span id="cb13-5"><a href="#cb13-5" tabindex="-1"></a><span class="co">#&gt; metadata(0):</span></span>
<span id="cb13-6"><a href="#cb13-6" tabindex="-1"></a><span class="co">#&gt; assays(1): counts</span></span>
<span id="cb13-7"><a href="#cb13-7" tabindex="-1"></a><span class="co">#&gt; rownames(100): ENSMUSG00000000001 ENSMUSG00000000028 ...</span></span>
<span id="cb13-8"><a href="#cb13-8" tabindex="-1"></a><span class="co">#&gt;   ENSMUSG00000000787 ENSMUSG00000000791</span></span>
<span id="cb13-9"><a href="#cb13-9" tabindex="-1"></a><span class="co">#&gt; rowData names(0):</span></span>
<span id="cb13-10"><a href="#cb13-10" tabindex="-1"></a><span class="co">#&gt; colnames(8): sample_1 sample_2 ... sample_7 sample_8</span></span>
<span id="cb13-11"><a href="#cb13-11" tabindex="-1"></a><span class="co">#&gt; colData names(2): sample group</span></span></code></pre></div>
<p>And we run sds_analyze.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a>sds <span class="ot">&lt;-</span> <span class="fu">sds_analyze</span>(sds)</span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a><span class="co">#&gt; sorting data 100% done...</span></span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a><span class="co">#&gt; t-test 100% done</span></span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb14-5"><a href="#cb14-5" tabindex="-1"></a><span class="co">#&gt; Welch test 100% done</span></span>
<span id="cb14-6"><a href="#cb14-6" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb14-7"><a href="#cb14-7" tabindex="-1"></a><span class="co">#&gt; Wilcox-test 100% done</span></span>
<span id="cb14-8"><a href="#cb14-8" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span></code></pre></div>
<p>And we can extract the results by doing:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">get_results</span>(sds)</span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a><span class="fu">head</span>(res)</span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a><span class="co">#&gt;                          Pval   test log2FoldChange_treated_vs_untreated</span></span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000000001 0.02895439 t-test                         -0.21245532</span></span>
<span id="cb15-5"><a href="#cb15-5" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000000056 0.88359469 t-test                          0.01577680</span></span>
<span id="cb15-6"><a href="#cb15-6" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000000058 0.18825794 t-test                         -0.36421697</span></span>
<span id="cb15-7"><a href="#cb15-7" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000000078 0.84664860 t-test                          0.02026956</span></span>
<span id="cb15-8"><a href="#cb15-8" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000000085 0.12619557 t-test                         -0.27483781</span></span>
<span id="cb15-9"><a href="#cb15-9" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000000088 0.82980304 t-test                         -0.02965276</span></span>
<span id="cb15-10"><a href="#cb15-10" tabindex="-1"></a><span class="co">#&gt;                    treated_mean untreated_mean</span></span>
<span id="cb15-11"><a href="#cb15-11" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000000001   1684.88769     1952.20932</span></span>
<span id="cb15-12"><a href="#cb15-12" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000000056    486.69980      481.40642</span></span>
<span id="cb15-13"><a href="#cb15-13" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000000058     67.23161       86.53937</span></span>
<span id="cb15-14"><a href="#cb15-14" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000000078   2658.42770     2621.33851</span></span>
<span id="cb15-15"><a href="#cb15-15" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000000085    180.73722      218.66639</span></span>
<span id="cb15-16"><a href="#cb15-16" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000000088    593.82628      606.15792</span></span></code></pre></div>
<p>You can access most results stored in sds with a function that starts
with get_ and ends with the name of the result. To see a full list of
results stored in sds, you can do this:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="fu">names</span>(sds<span class="sc">@</span>results)</span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a><span class="co">#&gt; [1] &quot;t_test_results&quot;      &quot;welch_test_results&quot;  &quot;wilcox_test_results&quot;</span></span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a><span class="co">#&gt; [4] &quot;all_results&quot;         &quot;ne&quot;                  &quot;nu&quot;                 </span></span>
<span id="cb16-4"><a href="#cb16-4" tabindex="-1"></a><span class="co">#&gt; [7] &quot;nn&quot;                  &quot;foldchange&quot;</span></span></code></pre></div>
<p>And any of these can be accessed by <a href="mailto:sds@results" class="email">sds@results</a>$result_name</p>
<p>Notice that we used the same function (sds_analyze) to analyze both
multi-group and two-group data. The sds_analyze function uses the design
information supplied during the creation of sds to decide whether to
apply two-group or multi-group tests.</p>
<p>We’re going to use the same sds_analyze function again for the next
part of this vignette.</p>
</div>
<div id="part3-multi-group-one-factor-design" class="section level2">
<h2>Part3: Multi-group one factor design</h2>
<p>The data we use here is some species level data from a metagenomic
analysis. There are 24 samples, and three experimental factors with 8
samples per group. For our purposes here, we’ll call the factors group1,
2, and 3. The pace of this analysis will be a bit faster than the
previous one, as most steps are very similar.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a><span class="co">#read in data</span></span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;species_abundance.csv&quot;</span>)</span>
<span id="cb17-3"><a href="#cb17-3" tabindex="-1"></a><span class="co">#fix rownames</span></span>
<span id="cb17-4"><a href="#cb17-4" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">cf</span>(x)</span>
<span id="cb17-5"><a href="#cb17-5" tabindex="-1"></a><span class="fu">head</span>(x)</span>
<span id="cb17-6"><a href="#cb17-6" tabindex="-1"></a><span class="co">#&gt;                                  sample_1 sample_2 sample_3 sample_4 sample_5</span></span>
<span id="cb17-7"><a href="#cb17-7" tabindex="-1"></a><span class="co">#&gt; Lachnospiraceae bacterium              14       33      245      224      167</span></span>
<span id="cb17-8"><a href="#cb17-8" tabindex="-1"></a><span class="co">#&gt; Dubosiella newyorkensis               583      181      118      145      112</span></span>
<span id="cb17-9"><a href="#cb17-9" tabindex="-1"></a><span class="co">#&gt; Parasutterella excrementihominis        4      241      431      327      343</span></span>
<span id="cb17-10"><a href="#cb17-10" tabindex="-1"></a><span class="co">#&gt; Barnesiella sp. CU968                 261      146      168      138      128</span></span>
<span id="cb17-11"><a href="#cb17-11" tabindex="-1"></a><span class="co">#&gt; Bacteroides thetaiotaomicron          356       74       79       91       95</span></span>
<span id="cb17-12"><a href="#cb17-12" tabindex="-1"></a><span class="co">#&gt; Lachnospiraceae bacterium 28-4          2        2     1176        7       92</span></span>
<span id="cb17-13"><a href="#cb17-13" tabindex="-1"></a><span class="co">#&gt;                                  sample_6 sample_7 sample_8 sample_9 sample_10</span></span>
<span id="cb17-14"><a href="#cb17-14" tabindex="-1"></a><span class="co">#&gt; Lachnospiraceae bacterium               9      996       71      312        85</span></span>
<span id="cb17-15"><a href="#cb17-15" tabindex="-1"></a><span class="co">#&gt; Dubosiella newyorkensis               118      665      131      121       117</span></span>
<span id="cb17-16"><a href="#cb17-16" tabindex="-1"></a><span class="co">#&gt; Parasutterella excrementihominis       21      128      241      367       364</span></span>
<span id="cb17-17"><a href="#cb17-17" tabindex="-1"></a><span class="co">#&gt; Barnesiella sp. CU968                 175      251      144      129       129</span></span>
<span id="cb17-18"><a href="#cb17-18" tabindex="-1"></a><span class="co">#&gt; Bacteroides thetaiotaomicron          155       80      129       76        83</span></span>
<span id="cb17-19"><a href="#cb17-19" tabindex="-1"></a><span class="co">#&gt; Lachnospiraceae bacterium 28-4          5     1170       43       19         4</span></span>
<span id="cb17-20"><a href="#cb17-20" tabindex="-1"></a><span class="co">#&gt;                                  sample_11 sample_12 sample_13 sample_14</span></span>
<span id="cb17-21"><a href="#cb17-21" tabindex="-1"></a><span class="co">#&gt; Lachnospiraceae bacterium               91      1731        22        12</span></span>
<span id="cb17-22"><a href="#cb17-22" tabindex="-1"></a><span class="co">#&gt; Dubosiella newyorkensis                113       210       120       460</span></span>
<span id="cb17-23"><a href="#cb17-23" tabindex="-1"></a><span class="co">#&gt; Parasutterella excrementihominis       246       255       393         8</span></span>
<span id="cb17-24"><a href="#cb17-24" tabindex="-1"></a><span class="co">#&gt; Barnesiella sp. CU968                  393       130        25        15</span></span>
<span id="cb17-25"><a href="#cb17-25" tabindex="-1"></a><span class="co">#&gt; Bacteroides thetaiotaomicron            75        82        77       371</span></span>
<span id="cb17-26"><a href="#cb17-26" tabindex="-1"></a><span class="co">#&gt; Lachnospiraceae bacterium 28-4          42       272         0         3</span></span>
<span id="cb17-27"><a href="#cb17-27" tabindex="-1"></a><span class="co">#&gt;                                  sample_15 sample_16 sample_17 sample_18</span></span>
<span id="cb17-28"><a href="#cb17-28" tabindex="-1"></a><span class="co">#&gt; Lachnospiraceae bacterium              221       289        11        35</span></span>
<span id="cb17-29"><a href="#cb17-29" tabindex="-1"></a><span class="co">#&gt; Dubosiella newyorkensis                117       105       114       737</span></span>
<span id="cb17-30"><a href="#cb17-30" tabindex="-1"></a><span class="co">#&gt; Parasutterella excrementihominis       386       120       385         7</span></span>
<span id="cb17-31"><a href="#cb17-31" tabindex="-1"></a><span class="co">#&gt; Barnesiella sp. CU968                  417       191       138       132</span></span>
<span id="cb17-32"><a href="#cb17-32" tabindex="-1"></a><span class="co">#&gt; Bacteroides thetaiotaomicron           119        91        78        76</span></span>
<span id="cb17-33"><a href="#cb17-33" tabindex="-1"></a><span class="co">#&gt; Lachnospiraceae bacterium 28-4         671        25         3         3</span></span>
<span id="cb17-34"><a href="#cb17-34" tabindex="-1"></a><span class="co">#&gt;                                  sample_19 sample_20 sample_21 sample_22</span></span>
<span id="cb17-35"><a href="#cb17-35" tabindex="-1"></a><span class="co">#&gt; Lachnospiraceae bacterium              168      1907        42       103</span></span>
<span id="cb17-36"><a href="#cb17-36" tabindex="-1"></a><span class="co">#&gt; Dubosiella newyorkensis                728       123       245       220</span></span>
<span id="cb17-37"><a href="#cb17-37" tabindex="-1"></a><span class="co">#&gt; Parasutterella excrementihominis         4       433         4       117</span></span>
<span id="cb17-38"><a href="#cb17-38" tabindex="-1"></a><span class="co">#&gt; Barnesiella sp. CU968                  123       116       213       284</span></span>
<span id="cb17-39"><a href="#cb17-39" tabindex="-1"></a><span class="co">#&gt; Bacteroides thetaiotaomicron            75       116       513       582</span></span>
<span id="cb17-40"><a href="#cb17-40" tabindex="-1"></a><span class="co">#&gt; Lachnospiraceae bacterium 28-4          14       461         3         5</span></span>
<span id="cb17-41"><a href="#cb17-41" tabindex="-1"></a><span class="co">#&gt;                                  sample_23 sample_24</span></span>
<span id="cb17-42"><a href="#cb17-42" tabindex="-1"></a><span class="co">#&gt; Lachnospiraceae bacterium              136        28</span></span>
<span id="cb17-43"><a href="#cb17-43" tabindex="-1"></a><span class="co">#&gt; Dubosiella newyorkensis                402        57</span></span>
<span id="cb17-44"><a href="#cb17-44" tabindex="-1"></a><span class="co">#&gt; Parasutterella excrementihominis        10         1</span></span>
<span id="cb17-45"><a href="#cb17-45" tabindex="-1"></a><span class="co">#&gt; Barnesiella sp. CU968                  123       420</span></span>
<span id="cb17-46"><a href="#cb17-46" tabindex="-1"></a><span class="co">#&gt; Bacteroides thetaiotaomicron            81       721</span></span>
<span id="cb17-47"><a href="#cb17-47" tabindex="-1"></a><span class="co">#&gt; Lachnospiraceae bacterium 28-4           8         4</span></span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="co">#make sample_info</span></span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a>sample_info <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">sample =</span> <span class="fu">colnames</span>(x), <span class="at">group =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;group1&quot;</span>, <span class="dv">8</span>), <span class="fu">rep</span>(<span class="st">&quot;group2&quot;</span>, <span class="dv">8</span>), <span class="fu">rep</span>(<span class="st">&quot;group3&quot;</span>, <span class="dv">8</span>)))</span>
<span id="cb18-3"><a href="#cb18-3" tabindex="-1"></a>sample_info</span>
<span id="cb18-4"><a href="#cb18-4" tabindex="-1"></a><span class="co">#&gt;       sample  group</span></span>
<span id="cb18-5"><a href="#cb18-5" tabindex="-1"></a><span class="co">#&gt; 1   sample_1 group1</span></span>
<span id="cb18-6"><a href="#cb18-6" tabindex="-1"></a><span class="co">#&gt; 2   sample_2 group1</span></span>
<span id="cb18-7"><a href="#cb18-7" tabindex="-1"></a><span class="co">#&gt; 3   sample_3 group1</span></span>
<span id="cb18-8"><a href="#cb18-8" tabindex="-1"></a><span class="co">#&gt; 4   sample_4 group1</span></span>
<span id="cb18-9"><a href="#cb18-9" tabindex="-1"></a><span class="co">#&gt; 5   sample_5 group1</span></span>
<span id="cb18-10"><a href="#cb18-10" tabindex="-1"></a><span class="co">#&gt; 6   sample_6 group1</span></span>
<span id="cb18-11"><a href="#cb18-11" tabindex="-1"></a><span class="co">#&gt; 7   sample_7 group1</span></span>
<span id="cb18-12"><a href="#cb18-12" tabindex="-1"></a><span class="co">#&gt; 8   sample_8 group1</span></span>
<span id="cb18-13"><a href="#cb18-13" tabindex="-1"></a><span class="co">#&gt; 9   sample_9 group2</span></span>
<span id="cb18-14"><a href="#cb18-14" tabindex="-1"></a><span class="co">#&gt; 10 sample_10 group2</span></span>
<span id="cb18-15"><a href="#cb18-15" tabindex="-1"></a><span class="co">#&gt; 11 sample_11 group2</span></span>
<span id="cb18-16"><a href="#cb18-16" tabindex="-1"></a><span class="co">#&gt; 12 sample_12 group2</span></span>
<span id="cb18-17"><a href="#cb18-17" tabindex="-1"></a><span class="co">#&gt; 13 sample_13 group2</span></span>
<span id="cb18-18"><a href="#cb18-18" tabindex="-1"></a><span class="co">#&gt; 14 sample_14 group2</span></span>
<span id="cb18-19"><a href="#cb18-19" tabindex="-1"></a><span class="co">#&gt; 15 sample_15 group2</span></span>
<span id="cb18-20"><a href="#cb18-20" tabindex="-1"></a><span class="co">#&gt; 16 sample_16 group2</span></span>
<span id="cb18-21"><a href="#cb18-21" tabindex="-1"></a><span class="co">#&gt; 17 sample_17 group3</span></span>
<span id="cb18-22"><a href="#cb18-22" tabindex="-1"></a><span class="co">#&gt; 18 sample_18 group3</span></span>
<span id="cb18-23"><a href="#cb18-23" tabindex="-1"></a><span class="co">#&gt; 19 sample_19 group3</span></span>
<span id="cb18-24"><a href="#cb18-24" tabindex="-1"></a><span class="co">#&gt; 20 sample_20 group3</span></span>
<span id="cb18-25"><a href="#cb18-25" tabindex="-1"></a><span class="co">#&gt; 21 sample_21 group3</span></span>
<span id="cb18-26"><a href="#cb18-26" tabindex="-1"></a><span class="co">#&gt; 22 sample_22 group3</span></span>
<span id="cb18-27"><a href="#cb18-27" tabindex="-1"></a><span class="co">#&gt; 23 sample_23 group3</span></span>
<span id="cb18-28"><a href="#cb18-28" tabindex="-1"></a><span class="co">#&gt; 24 sample_24 group3</span></span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a><span class="co">#make sds</span></span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a>sds <span class="ot">&lt;-</span> <span class="fu">make_sds</span>(x, <span class="at">colData =</span> sample_info, <span class="at">design =</span> <span class="sc">~</span> group)</span>
<span id="cb19-3"><a href="#cb19-3" tabindex="-1"></a>sds <span class="ot">&lt;-</span> <span class="fu">sds_analyze</span>(sds)</span>
<span id="cb19-4"><a href="#cb19-4" tabindex="-1"></a><span class="co">#&gt; sorting data 100% done...</span></span>
<span id="cb19-5"><a href="#cb19-5" tabindex="-1"></a><span class="co">#&gt; Using one way anova </span></span>
<span id="cb19-6"><a href="#cb19-6" tabindex="-1"></a><span class="co">#&gt; anova 100% done</span></span>
<span id="cb19-7"><a href="#cb19-7" tabindex="-1"></a><span class="co">#&gt; Tukey test 100% done</span></span>
<span id="cb19-8"><a href="#cb19-8" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb19-9"><a href="#cb19-9" tabindex="-1"></a><span class="co">#&gt; Welch test 100% done</span></span>
<span id="cb19-10"><a href="#cb19-10" tabindex="-1"></a><span class="co">#&gt; DunnettT3 test 100% done</span></span>
<span id="cb19-11"><a href="#cb19-11" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span>
<span id="cb19-12"><a href="#cb19-12" tabindex="-1"></a><span class="co">#&gt; Kruskal-Wallis Test 63% done...Kruskal-Wallis Test 100% done</span></span>
<span id="cb19-13"><a href="#cb19-13" tabindex="-1"></a><span class="co">#&gt; Dunn Test 63% done...Dunn Test 100% done</span></span>
<span id="cb19-14"><a href="#cb19-14" tabindex="-1"></a><span class="co">#&gt; Calculating fold changes...</span></span></code></pre></div>
<p>Notice here that statomatic tells us that it’s using a one way anova
to analyze the normal-equal variance portion of the data. This is
because of our specified design.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">get_results</span>(sds)</span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a><span class="fu">head</span>(res)</span>
<span id="cb20-3"><a href="#cb20-3" tabindex="-1"></a><span class="co">#&gt;                                          Pval Pval_group1_vs_group2</span></span>
<span id="cb20-4"><a href="#cb20-4" tabindex="-1"></a><span class="co">#&gt; Parasutterella excrementihominis 0.1991824896             0.8062039</span></span>
<span id="cb20-5"><a href="#cb20-5" tabindex="-1"></a><span class="co">#&gt; Muricaecibacterium torontonense  0.0681265365             0.5464864</span></span>
<span id="cb20-6"><a href="#cb20-6" tabindex="-1"></a><span class="co">#&gt; Muribaculaceae bacterium         0.0010328865             0.4923547</span></span>
<span id="cb20-7"><a href="#cb20-7" tabindex="-1"></a><span class="co">#&gt; Sangeribacter muris              0.3985184169             0.8117231</span></span>
<span id="cb20-8"><a href="#cb20-8" tabindex="-1"></a><span class="co">#&gt; Anaerotruncus colihominis        0.6083336357             0.7787006</span></span>
<span id="cb20-9"><a href="#cb20-9" tabindex="-1"></a><span class="co">#&gt; Bacteroides sp. CAG:927          0.0009232984             1.0000000</span></span>
<span id="cb20-10"><a href="#cb20-10" tabindex="-1"></a><span class="co">#&gt;                                  log2FoldChange_group1_vs_group2</span></span>
<span id="cb20-11"><a href="#cb20-11" tabindex="-1"></a><span class="co">#&gt; Parasutterella excrementihominis                      -0.3011695</span></span>
<span id="cb20-12"><a href="#cb20-12" tabindex="-1"></a><span class="co">#&gt; Muricaecibacterium torontonense                       -0.5635121</span></span>
<span id="cb20-13"><a href="#cb20-13" tabindex="-1"></a><span class="co">#&gt; Muribaculaceae bacterium                               0.6818240</span></span>
<span id="cb20-14"><a href="#cb20-14" tabindex="-1"></a><span class="co">#&gt; Sangeribacter muris                                    0.2115041</span></span>
<span id="cb20-15"><a href="#cb20-15" tabindex="-1"></a><span class="co">#&gt; Anaerotruncus colihominis                              0.4340958</span></span>
<span id="cb20-16"><a href="#cb20-16" tabindex="-1"></a><span class="co">#&gt; Bacteroides sp. CAG:927                                0.0000000</span></span>
<span id="cb20-17"><a href="#cb20-17" tabindex="-1"></a><span class="co">#&gt;                                  Pval_group1_vs_group3</span></span>
<span id="cb20-18"><a href="#cb20-18" tabindex="-1"></a><span class="co">#&gt; Parasutterella excrementihominis           0.461211401</span></span>
<span id="cb20-19"><a href="#cb20-19" tabindex="-1"></a><span class="co">#&gt; Muricaecibacterium torontonense            0.357348922</span></span>
<span id="cb20-20"><a href="#cb20-20" tabindex="-1"></a><span class="co">#&gt; Muribaculaceae bacterium                   0.014194068</span></span>
<span id="cb20-21"><a href="#cb20-21" tabindex="-1"></a><span class="co">#&gt; Sangeribacter muris                        0.727643346</span></span>
<span id="cb20-22"><a href="#cb20-22" tabindex="-1"></a><span class="co">#&gt; Anaerotruncus colihominis                  0.949127148</span></span>
<span id="cb20-23"><a href="#cb20-23" tabindex="-1"></a><span class="co">#&gt; Bacteroides sp. CAG:927                    0.002511996</span></span>
<span id="cb20-24"><a href="#cb20-24" tabindex="-1"></a><span class="co">#&gt;                                  log2FoldChange_group1_vs_group3</span></span>
<span id="cb20-25"><a href="#cb20-25" tabindex="-1"></a><span class="co">#&gt; Parasutterella excrementihominis                       0.8531586</span></span>
<span id="cb20-26"><a href="#cb20-26" tabindex="-1"></a><span class="co">#&gt; Muricaecibacterium torontonense                        1.4364879</span></span>
<span id="cb20-27"><a href="#cb20-27" tabindex="-1"></a><span class="co">#&gt; Muribaculaceae bacterium                              -1.0093379</span></span>
<span id="cb20-28"><a href="#cb20-28" tabindex="-1"></a><span class="co">#&gt; Sangeribacter muris                                   -0.2250666</span></span>
<span id="cb20-29"><a href="#cb20-29" tabindex="-1"></a><span class="co">#&gt; Anaerotruncus colihominis                             -0.1610624</span></span>
<span id="cb20-30"><a href="#cb20-30" tabindex="-1"></a><span class="co">#&gt; Bacteroides sp. CAG:927                               -1.6455040</span></span>
<span id="cb20-31"><a href="#cb20-31" tabindex="-1"></a><span class="co">#&gt;                                  Pval_group2_vs_group3</span></span>
<span id="cb20-32"><a href="#cb20-32" tabindex="-1"></a><span class="co">#&gt; Parasutterella excrementihominis           0.182048409</span></span>
<span id="cb20-33"><a href="#cb20-33" tabindex="-1"></a><span class="co">#&gt; Muricaecibacterium torontonense            0.055825056</span></span>
<span id="cb20-34"><a href="#cb20-34" tabindex="-1"></a><span class="co">#&gt; Muribaculaceae bacterium                   0.000975412</span></span>
<span id="cb20-35"><a href="#cb20-35" tabindex="-1"></a><span class="co">#&gt; Sangeribacter muris                        0.366905054</span></span>
<span id="cb20-36"><a href="#cb20-36" tabindex="-1"></a><span class="co">#&gt; Anaerotruncus colihominis                  0.593457381</span></span>
<span id="cb20-37"><a href="#cb20-37" tabindex="-1"></a><span class="co">#&gt; Bacteroides sp. CAG:927                    0.002511996</span></span>
<span id="cb20-38"><a href="#cb20-38" tabindex="-1"></a><span class="co">#&gt;                                  log2FoldChange_group2_vs_group3 group1_mean</span></span>
<span id="cb20-39"><a href="#cb20-39" tabindex="-1"></a><span class="co">#&gt; Parasutterella excrementihominis                       1.1543281     217.000</span></span>
<span id="cb20-40"><a href="#cb20-40" tabindex="-1"></a><span class="co">#&gt; Muricaecibacterium torontonense                        2.0000000     163.750</span></span>
<span id="cb20-41"><a href="#cb20-41" tabindex="-1"></a><span class="co">#&gt; Muribaculaceae bacterium                              -1.6911619      38.500</span></span>
<span id="cb20-42"><a href="#cb20-42" tabindex="-1"></a><span class="co">#&gt; Sangeribacter muris                                   -0.4365707      19.250</span></span>
<span id="cb20-43"><a href="#cb20-43" tabindex="-1"></a><span class="co">#&gt; Anaerotruncus colihominis                             -0.5951583      15.875</span></span>
<span id="cb20-44"><a href="#cb20-44" tabindex="-1"></a><span class="co">#&gt; Bacteroides sp. CAG:927                               -1.6455040       8.750</span></span>
<span id="cb20-45"><a href="#cb20-45" tabindex="-1"></a><span class="co">#&gt;                                  group2_mean group3_mean       test</span></span>
<span id="cb20-46"><a href="#cb20-46" tabindex="-1"></a><span class="co">#&gt; Parasutterella excrementihominis     267.375     120.125 tukey-test</span></span>
<span id="cb20-47"><a href="#cb20-47" tabindex="-1"></a><span class="co">#&gt; Muricaecibacterium torontonense      242.000      60.500 tukey-test</span></span>
<span id="cb20-48"><a href="#cb20-48" tabindex="-1"></a><span class="co">#&gt; Muribaculaceae bacterium              24.000      77.500 tukey-test</span></span>
<span id="cb20-49"><a href="#cb20-49" tabindex="-1"></a><span class="co">#&gt; Sangeribacter muris                   16.625      22.500 tukey-test</span></span>
<span id="cb20-50"><a href="#cb20-50" tabindex="-1"></a><span class="co">#&gt; Anaerotruncus colihominis             11.750      17.750 tukey-test</span></span>
<span id="cb20-51"><a href="#cb20-51" tabindex="-1"></a><span class="co">#&gt; Bacteroides sp. CAG:927                8.750      27.375 tukey-test</span></span></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
